<!doctype html><html lang="en">
 <head>
  <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
  <title>GDI Harmonised Minimal Data Model</title>
  <meta content="width=device-width, initial-scale=1, shrink-to-fit=no" name="viewport">
<style data-fill-with="stylesheet">/******************************************************************************
 *                   Style sheet for the W3C specifications                   *
 *
 * Special classes handled by this style sheet include:
 *
 * Indices
 *   - .toc for the Table of Contents (<ol class="toc">)
 *     + <span class="secno"> for the section numbers
 *   - #toc for the Table of Contents (<nav id="toc">)
 *   - ul.index for Indices (<a href="#ref">term</a><span>, in § N.M</span>)
 *   - table.index for Index Tables (e.g. for properties or elements)
 *
 * Structural Markup
 *   - table.data for general data tables
 *     -> use 'scope' attribute, <colgroup>, <thead>, and <tbody> for best results !
 *     -> use <table class='complex data'> for extra-complex tables
 *     -> use <td class='long'> for paragraph-length cell content
 *     -> use <td class='pre'> when manual line breaks/indentation would help readability
 *   - dl.switch for switch statements
 *   - ol.algorithm for algorithms (helps to visualize nesting)
 *   - .figure and .caption (HTML4) and figure and figcaption (HTML5)
 *     -> .sidefigure for right-floated figures
 *   - ins/del
 *     -> ins/del.c### for candidate and proposed changes (amendments)
 *
 * Code
 *   - pre and code
 *
 * Special Sections
 *   - .note       for informative notes             (div, p, span, aside, details)
 *   - .example    for informative examples          (div, p, pre, span)
 *   - .issue      for issues                        (div, p, span)
 *   - .advisement for loud normative statements     (div, p, strong)
 *   - .annoying-warning for spec obsoletion notices (div, aside, details)
 *   - .correction for "candidate corrections"       (div, aside, details, section)
 *   - .addition   for "candidate additions"         (div, aside, details, section)
 *   - .correction.proposed for "proposed corrections" (div, aside, details, section)
 *   - .addition.proposed   for "proposed additions"   (div, aside, details, section)
 *
 * Definition Boxes
 *   - pre.def   for WebIDL definitions
 *   - table.def for tables that define other entities (e.g. CSS properties)
 *   - dl.def    for definition lists that define other entitles (e.g. HTML elements)
 *
 * Numbering
 *   - .secno for section numbers in .toc and headings (<span class='secno'>3.2</span>)
 *   - .marker for source-inserted example/figure/issue numbers (<span class='marker'>Issue 4</span>)
 *   - ::before styled for CSS-generated issue/example/figure numbers:
 *     -> Documents wishing to use this only need to add
 *        figcaption::before,
 *        .caption::before { content: "Figure "  counter(figure) " ";  }
 *        .example::before { content: "Example " counter(example) " "; }
 *        .issue::before   { content: "Issue "   counter(issue) " ";   }
 *
 * Header Stuff (ignore, just don't conflict with these classes)
 *   - .head for the header
 *   - .copyright for the copyright
 *
 * Outdated warning for old specs
 *
 * Miscellaneous
 *   - .overlarge for things that should be as wide as possible, even if
 *     that overflows the body text area. This can be used on an item or
 *     on its container, depending on the effect desired.
 *     Note that this styling basically doesn't help at all when printing,
 *     since A4 paper isn't much wider than the max-width here.
 *     It's better to design things to fit into a narrower measure if possible.
 *
 *   - js-added ToC jump links (see fixup.js)
 *
 ******************************************************************************/

/* color variables included separately for reliability */

/******************************************************************************/
/*                                    Body                                    */
/******************************************************************************/

	html {
	}

	body {
		counter-reset: example figure issue;

		/* Layout */
		max-width: 50em;			  /* limit line length to 50em for readability   */
		margin: 0 auto;				/* center text within page                    */
		padding: 1.6em 1.5em 2em 50px; /* assume 16px font size for downlevel clients */
		padding: 1.6em 1.5em 2em calc(26px + 1.5em); /* leave space for status flag    */

		/* Typography */
		line-height: 1.5;
		font-family: sans-serif;
		widows: 2;
		orphans: 2;
		word-wrap: break-word;
		overflow-wrap: break-word;
		hyphens: auto;

		color: black;
		color: var(--text);
		background: white top left fixed no-repeat;
		background: var(--bg) top left fixed no-repeat;
		background-size: 25px auto;
	}


/******************************************************************************/
/*                         Front Matter & Navigation                          */
/******************************************************************************/

/** Header ********************************************************************/

	div.head { margin-bottom: 1em; }
	div.head hr { border-style: solid; }

	div.head h1 {
		font-weight: bold;
		margin: 0 0 .1em;
		font-size: 220%;
	}

	div.head h2 { margin-bottom: 1.5em;}

/** W3C Logo ******************************************************************/

	.head .logo {
		float: right;
		margin: 0.4rem 0 0.2rem .4rem;
	}

	.head img[src*="logos/W3C"] {
		display: block;
		border: solid #1a5e9a;
		border: solid var(--logo-bg);
		border-width: .65rem .7rem .6rem;
		border-radius: .4rem;
		background: #1a5e9a;
		background: var(--logo-bg);
		color: white;
		color: var(--logo-text);
		font-weight: bold;
	}

	.head a:hover > img[src*="logos/W3C"],
	.head a:focus > img[src*="logos/W3C"] {
		opacity: .8;
	}

	.head a:active > img[src*="logos/W3C"] {
		background: #c00;
		background: var(--logo-active-bg);
		border-color: #c00;
		border-color: var(--logo-active-bg);
	}

	/* see also additional rules in Link Styling section */

/** Copyright *****************************************************************/

	p.copyright,
	p.copyright small { font-size: small; }

/** Back to Top / ToC Toggle **************************************************/

	@media print {
		#toc-nav {
			display: none;
		}
	}
	@media not print {
		#toc-nav {
			position: fixed;
			z-index: 3;
			bottom: 0; left: 0;
			margin: 0;
			min-width: 1.33em;
			border-top-right-radius: 2rem;
			box-shadow: 0 0 2px;
			font-size: 1.5em;
		}
		#toc-nav > a {
			display: block;
			white-space: nowrap;

			height: 1.33em;
			padding: .1em 0.3em;
			margin: 0;

			box-shadow: 0 0 2px;
			border: none;
			border-top-right-radius: 1.33em;

			color: #707070;
			color: var(--tocnav-normal-text);
			background: white;
			background: var(--tocnav-normal-bg);
		}
		#toc-nav > a:hover,
		#toc-nav > a:focus {
			color: black;
			color: var(--tocnav-hover-text);
			background: #f8f8f8;
			background: var(--tocnav-hover-bg);
		}
		#toc-nav > a:active {
			color: #c00;
			color: var(--tocnav-active-text);
			background: white;
			background: var(--tocnav-active-bg);
		}

		#toc-nav > #toc-jump {
			padding-bottom: 2em;
			margin-bottom: -1.9em;
		}

		/* statusbar gets in the way on keyboard focus; remove once browsers fix */
		#toc-nav > a[href="#toc"]:not(:hover):focus:last-child {
			padding-bottom: 1.5rem;
		}

		#toc-nav:not(:hover) > a:not(:focus) > span + span {
			/* Ideally this uses :focus-within on #toc-nav */
			display: none;
		}
		#toc-nav > a > span + span {
			padding-right: 0.2em;
		}
	}

/** ToC Sidebar ***************************************************************/

	/* Floating sidebar */
	@media screen {
		body.toc-sidebar #toc {
			position: fixed;
			top: 0; bottom: 0;
			left: 0;
			width: 23.5em;
			max-width: 80%;
			max-width: calc(100% - 2em - 26px);
			overflow: auto;
			padding: 0 1em;
			padding-left: 42px;
			padding-left: calc(1em + 26px);
			color: black;
			color: var(--tocsidebar-text);
			background: inherit;
			background-color: #f7f8f9;
			background-color: var(--tocsidebar-bg);
			z-index: 1;
			box-shadow: -.1em 0 .25em rgba(0,0,0,.1) inset;
			box-shadow: -.1em 0 .25em var(--tocsidebar-shadow) inset;
		}
		body.toc-sidebar #toc h2 {
			margin-top: .8rem;
			font-variant: small-caps;
			font-variant: all-small-caps;
			text-transform: lowercase;
			font-weight: bold;
			color: gray;
			color: hsla(203,20%,40%,.7);
			color: var(--tocsidebar-heading-text);
		}
		body.toc-sidebar #toc-jump:not(:focus) {
			width: 0;
			height: 0;
			padding: 0;
			position: absolute;
			overflow: hidden;
		}
	}
	/* Hide main scroller when only the ToC is visible anyway */
	@media screen and (max-width: 28em) {
		body.toc-sidebar {
			overflow: hidden;
		}
	}

	/* Sidebar with its own space */
	@media screen and (min-width: 78em) {
		body:not(.toc-inline) #toc {
			position: fixed;
			top: 0; bottom: 0;
			left: 0;
			width: 23.5em;
			overflow: auto;
			padding: 0 1em;
			padding-left: 42px;
			padding-left: calc(1em + 26px);
			color: black;
			color: var(--tocsidebar-text);
			background: inherit;
			background-color: #f7f8f9;
			background-color: var(--tocsidebar-bg);
			z-index: 1;
			box-shadow: -.1em 0 .25em rgba(0,0,0,.1) inset;
			box-shadow: -.1em 0 .25em var(--tocsidebar-shadow) inset;
		}
		body:not(.toc-inline) #toc h2 {
			margin-top: .8rem;
			font-variant: small-caps;
			font-variant: all-small-caps;
			text-transform: lowercase;
			font-weight: bold;
			color: gray;
			color: hsla(203,20%,40%,.7);
			color: var(--tocsidebar-heading-text);
		}

		body:not(.toc-inline) {
			padding-left: 29em;
		}
		/* See also Overflow section at the bottom */

		body:not(.toc-inline) #toc-jump:not(:focus) {
			width: 0;
			height: 0;
			padding: 0;
			position: absolute;
			overflow: hidden;
		}
	}
	@media screen and (min-width: 90em) {
		body:not(.toc-inline) {
			margin: 0 4em;
		}
	}

/******************************************************************************/
/*                                Sectioning                                  */
/******************************************************************************/

/** Headings ******************************************************************/

	h1, h2, h3, h4, h5, h6, dt {
		page-break-after: avoid;
		page-break-inside: avoid;
		font: 100% sans-serif;   /* Reset all font styling to clear out UA styles */
		font-family: inherit;	/* Inherit the font family. */
		line-height: 1.2;		/* Keep wrapped headings compact */
		hyphens: manual;		/* Hyphenated headings look weird */
	}

	h2, h3, h4, h5, h6 {
		margin-top: 3rem;
	}

	h1, h2, h3 {
		color: #005A9C;
		color: var(--heading-text);
	}

	h1 { font-size: 170%; }
	h2 { font-size: 140%; }
	h3 { font-size: 120%; }
	h4 { font-weight: bold; }
	h5 { font-style: italic; }
	h6 { font-variant: small-caps; }
	dt { font-weight: bold; }

/** Subheadings ***************************************************************/

	h1 + h2,
	#profile-and-date {
		/* #profile-and-date is a subtitle in an H2 under the H1 */
		margin-top: 0;
	}
	h2 + h3,
	h3 + h4,
	h4 + h5,
	h5 + h6 {
		margin-top: 1.2em; /* = 1 x line-height */
	}

/** Section divider ***********************************************************/

	:not(.head) > :not(.head) + hr {
		font-size: 1.5em;
		text-align: center;
		margin: 1em auto;
		height: auto;
		color: black;
		color: var(--hr-text);
		border: transparent solid 0;
		background: transparent;
	}
	:not(.head) > hr::before {
		content: "\2727\2003\2003\2727\2003\2003\2727";
	}

/******************************************************************************/
/*                            Paragraphs and Lists                            */
/******************************************************************************/

	p {
		margin: 1em 0;
	}

	dd > p:first-child,
	li > p:first-child {
		margin-top: 0;
	}

	ul, ol {
		margin-left: 0;
		padding-left: 2em;
	}

	li {
		margin: 0.25em 0 0.5em;
		padding: 0;
	}

	dl dd {
		margin: 0 0 .5em 2em;
	}

	.head dd + dd { /* compact for header */
		margin-top: -.5em;
	}

	/* Style for algorithms */
	ol.algorithm ol:not(.algorithm),
	.algorithm > ol ol:not(.algorithm) {
	border-left: 0.5em solid #DEF;
	border-left: 0.5em solid var(--algo-border);
	}

	/* Put nice boxes around each algorithm. */
	[data-algorithm]:not(.heading) {
	 padding: .5em;
	 border: thin solid #ddd;
	 border: thin solid var(--algo-border);
	 border-radius: .5em;
	 margin: .5em calc(-0.5em - 1px);
	}
	[data-algorithm]:not(.heading) > :first-child {
	 margin-top: 0;
	}
	[data-algorithm]:not(.heading) > :last-child {
	 margin-bottom: 0;
	}

	/* Style for switch/case <dl>s */
	dl.switch > dd > ol.only,
	dl.switch > dd > .only > ol {
	margin-left: 0;
	}
	dl.switch > dd > ol.algorithm,
	dl.switch > dd > .algorithm > ol {
	margin-left: -2em;
	}
	dl.switch {
	padding-left: 2em;
	}
	dl.switch > dt {
	text-indent: -1.5em;
	margin-top: 1em;
	}
	dl.switch > dt + dt {
	margin-top: 0;
	}
	dl.switch > dt::before {
	content: '\21AA';
	padding: 0 0.5em 0 0;
	display: inline-block;
	width: 1em;
	text-align: right;
	line-height: 0.5em;
	}

/** Terminology Markup ********************************************************/


/******************************************************************************/
/*                                 Inline Markup                              */
/******************************************************************************/

/** Terminology Markup ********************************************************/
	dfn   { /* Defining instance */
		font-weight: bolder;
	}
	a > i { /* Instance of term */
		font-style: normal;
	}
	dt dfn code, code.idl {
		font-size: inherit;
	}
	dfn var {
		font-style: normal;
	}

/** Change Marking ************************************************************/

	del {
		color: #aa0000;
		color: var(--del-text);
		background: transparent;
		background: var(--del-bg);
		text-decoration: line-through;
	}
	ins {
		color: #006100;
		color: var(--ins-text);
		background: transparent;
		background: var(--ins-bg);
		text-decoration: underline;
	}

	/* for amendments (candidate/proposed changes) */

	.amendment ins, .correction ins, .addition ins,
	ins[class^=c] {
		text-decoration-style: dotted;
	}
	.amendment del, .correction del, .addition del,
	del[class^=c] {
		text-decoration-style: dotted;
	}
	.amendment.proposed ins, .correction.proposed ins, .addition.proposed ins,
	ins[class^=c].proposed {
		text-decoration-style: double;
	}
	.amendment.proposed del, .correction.proposed del, .addition.proposed del,
	del[class^=c].proposed {
		text-decoration-style: double;
	}

/** Miscellaneous improvements to inline formatting ***************************/

	sup {
		vertical-align: super;
		font-size: 80%
	}

/******************************************************************************/
/*                                    Code                                    */
/******************************************************************************/

/** General monospace/pre rules ***********************************************/

	pre, code, samp {
		font-family: Menlo, Consolas, "DejaVu Sans Mono", Monaco, monospace;
		font-size: .9em;
		hyphens: none;
		text-transform: none;
		text-align: left;
		text-align: start;
		font-variant: normal;
		orphans: 3;
		widows: 3;
		page-break-before: avoid;
	}
	pre code,
	code code {
		font-size: 100%;
	}

	pre {
		margin-top: 1em;
		margin-bottom: 1em;
		overflow: auto;
	}

/** Inline Code fragments *****************************************************/

	/* Do something nice. */

/******************************************************************************/
/*                                    Links                                   */
/******************************************************************************/

/** General Hyperlinks ********************************************************/

	/* We hyperlink a lot, so make it less intrusive */
	a[href] {
		color: #034575;
		color: var(--a-normal-text);
		text-decoration: underline #707070;
		text-decoration: underline var(--a-normal-underline);
		text-decoration-skip-ink: none;
	}
	a:visited {
		color: #034575;
		color: var(--a-visited-text);
		text-decoration-color: #bbb;
		text-decoration-color: var(--a-visited-underline);
	}

	/* Indicate interaction with the link */
	a[href]:focus,
	a[href]:hover {
		text-decoration-thickness: 2px;
	}
	a[href]:active {
		color: #c00;
		color: var(--a-active-text);
		text-decoration-color: #c00;
		text-decoration-color: var(--a-active-underline);
	}

	/* Backout above styling for W3C logo */
	.head .logo,
	.head .logo a {
		border: none;
		text-decoration: none;
		background: transparent;
	}

/******************************************************************************/
/*                                    Images                                  */
/******************************************************************************/

	img {
		border-style: none;
	}

	img, svg {
		/* Intentionally not color-scheme aware. */
		background: white;
	}

	/* For autogen numbers, add
	  .caption::before, figcaption::before { content: "Figure " counter(figure) ". "; }
	*/

	figure, .figure, .sidefigure {
		page-break-inside: avoid;
		text-align: center;
		margin: 2.5em 0;
	}
	.figure img,	.sidefigure img,	figure img,
	.figure object, .sidefigure object, figure object {
		max-width: 100%;
		margin: auto;
		height: auto;
	}
	.figure pre, .sidefigure pre, figure pre {
		text-align: left;
		display: table;
		margin: 1em auto;
	}
	.figure table, figure table {
		margin: auto;
	}
	@media screen and (min-width: 20em) {
		.sidefigure {
			float: right;
			width: 50%;
			margin: 0 0 0.5em 0.5em;
		}
	}
	.caption, figcaption, caption {
		font-style: italic;
		font-size: 90%;
	}
	.caption::before, figcaption::before, figcaption > .marker {
		font-weight: bold;
	}
	.caption, figcaption {
		counter-increment: figure;
	}

	/* DL list is indented 2em, but figure inside it is not */
	dd > .figure, dd > figure { margin-left: -2em; }

/******************************************************************************/
/*                             Colored Boxes                                  */
/******************************************************************************/

	.issue, .note, .example, .assertion, .advisement, blockquote,
	.amendment, .correction, .addition {
		margin: 1em auto;
		padding: .5em;
		border: .5em;
		border-left-style: solid;
		page-break-inside: avoid;
	}
	span.issue, span.note {
		padding: .1em .5em .15em;
		border-right-style: solid;
	}

	blockquote > :first-child,
	.note  > p:first-child,
	.issue > p:first-child,
	.amendment > p:first-child,
	.correction > p:first-child,
	.addition > p:first-child {
		margin-top: 0;
	}
	blockquote > :last-child,
	.note  > p:last-child,
	.issue > p:last-child,
	.amendment > p:last-child,
	.correction > p:last-child,
	.addition > p:last-child {
		margin-bottom: 0;
	}


	.issue::before, .issue > .marker,
	.example::before, .example > .marker,
	.note::before, .note > .marker,
	details.note > summary > .marker,
	.amendment::before, .amendment > .marker,
	details.amendment > summary > .marker,
	.addition::before, .addition > .marker,
	addition.amendment > summary > .marker,
	.correction::before, .correction > .marker,
	correction.amendment > summary > .marker
	{
		text-transform: uppercase;
		padding-right: 1em;
	}

	.example::before, .example > .marker {
		display: block;
		padding-right: 0em;
	}

/** Blockquotes ***************************************************************/

	blockquote {
		border-color: silver;
		border-color: var(--blockquote-border);
		background: transparent;
		background: var(--blockquote-bg);
		color: currentcolor;
		color: var(--blockquote-text);
	}

/** Open issue ****************************************************************/

	.issue {
		border-color: #e05252;
		border-color: var(--issue-border);
		background: #fbe9e9;
		background: var(--issue-bg);
		color: black;
		color: var(--issue-text);
		counter-increment: issue;
		overflow: auto;
	}
	.issue::before, .issue > .marker {
		color: #831616;
		color: var(--issueheading-text);
	}
	/* Add .issue::before { content: "Issue " counter(issue) " "; } for autogen numbers,
	  or use class="marker" to mark up the issue number in source. */

/** Example *******************************************************************/

	.example {
		border-color: #e0cb52;
		border-color: var(--example-border);
		background: #fcfaee;
		background: var(--example-bg);
		color: black;
		color: var(--example-text);
		counter-increment: example;
		overflow: auto;
		clear: both;
	}
	.example::before, .example > .marker {
		color: #574b0f;
		color: var(--exampleheading-text);
	}
	/* Add .example::before { content: "Example " counter(example) " "; } for autogen numbers,
	  or use class="marker" to mark up the example number in source. */

/** Non-normative Note ********************************************************/

	.note {
		border-color: #52e052;
		border-color: var(--note-border);
		background: #e9fbe9;
		background: var(--note-bg);
		color: black;
		color: var(--note-text);
		overflow: auto;
	}

	.note::before, .note > .marker,
	details.note > summary {
		color: hsl(120, 70%, 30%);
		color: var(--noteheading-text);
	}
	/* Add .note::before { content: "Note "; } for autogen label,
	  or use class="marker" to mark up the label in source. */

	details.note[open] > summary {
		border-bottom: 1px silver solid;
		border-bottom: 1px var(--notesummary-underline) solid;
	}

/** Assertion Box *************************************************************/
	/*  for assertions in algorithms */

	.assertion {
		border-color: #AAA;
		border-color: var(--assertion-border);
		background: #EEE;
		background: var(--assertion-bg);
		color: black;
		color: var(--assertion-text);
	}

/** Advisement Box ************************************************************/
	/*  for attention-grabbing normative statements */

	.advisement {
		border-color: orange;
		border-color: var(--advisement-border);
		border-style: none solid;
		background: #fec;
		background: var(--advisement-bg);
		color: black;
		color: var(--advisement-text);
	}
	strong.advisement {
		display: block;
		text-align: center;
	}
	.advisement::before, .advisement > .marker {
		color: #b35f00;
		color: var(--advisementheading-text);
	}

/** Amendment Box *************************************************************/

	.amendment, .correction, .addition {
		border-color: #330099;
		border-color: var(--amendment-border);
		background: #F5F0FF;
		background: var(--amendment-bg);
		color: black;
		color: var(--amendment-text);
	}
	.amendment.proposed, .correction.proposed, .addition.proposed {
		border-style: solid;
		border-block-width: 0.25em;
	}
	.amendment::before, .amendment > .marker,
	details.amendment > summary::before, details.amendment > summary > .marker,
	.correction::before, .correction > .marker,
	details.correction > summary::before, details.correction > summary > .marker,
	.addition::before, .addition > .marker,
	details.addition > summary::before, details.addition > summary > .marker {
		color: #220066;
		color: var(--amendmentheading-text);
	}
	.amendment.proposed::before, .amendment.proposed > .marker,
	details.amendment.proposed > summary::before, details.amendment.proposed > summary > .marker,
	.correction.proposed::before, .correction.proposed > .marker,
	details.correction.proposed > summary::before, details.correction.proposed > summary > .marker,
	.addition.proposed::before, .addition.proposed > .marker,
	details.addition.proposed > summary::before, details.addition.proposed > summary > .marker {
		font-weight: bold;
	}

/** Spec Obsoletion Notice ****************************************************/
	/* obnoxious obsoletion notice for older/abandoned specs. */

	details {
		display: block;
	}
	summary {
		font-weight: bolder;
	}

	.annoying-warning:not(details),
	details.annoying-warning:not([open]) > summary,
	details.annoying-warning[open] {
		background: hsla(40,100%,50%,0.95);
		background: var(--warning-bg);
		color: black;
		color: var(--warning-text);
		padding: .75em 1em;
		border: red;
		border: var(--warning-border);
		border-style: solid none;
		box-shadow: 0 2px 8px black;
		text-align: center;
	}
	.annoying-warning :last-child {
		margin-bottom: 0;
	}

@media not print {
	details.annoying-warning[open] {
		position: fixed;
		left: 0;
		right: 0;
		bottom: 2em;
		z-index: 1000;
	}
}

	details.annoying-warning:not([open]) > summary {
		text-align: center;
	}

/** Entity Definition Boxes ***************************************************/

	.def {
		padding: .5em 1em;
		background: #def;
		background: var(--def-bg);
		margin: 1.2em 0;
		border-left: 0.5em solid #8ccbf2;
		border-left: 0.5em solid var(--def-border);
		color: black;
		color: var(--def-text);
	}

/******************************************************************************/
/*                                    Tables                                  */
/******************************************************************************/

	th, td {
		text-align: left;
		text-align: start;
	}

/** Property/Descriptor Definition Tables *************************************/

	table.def {
		/* inherits .def box styling, see above */
		width: 100%;
		border-spacing: 0;
	}

	table.def td,
	table.def th {
		padding: 0.5em;
		vertical-align: baseline;
		border-bottom: 1px solid #bbd7e9;
		border-bottom: 1px solid var(--defrow-border);
	}

	table.def > tbody > tr:last-child th,
	table.def > tbody > tr:last-child td {
		border-bottom: 0;
	}

	table.def th {
		font-style: italic;
		font-weight: normal;
		padding-left: 1em;
		width: 3em;
	}

	/* For when values are extra-complex and need formatting for readability */
	table td.pre {
		white-space: pre-wrap;
	}

	/* A footnote at the bottom of a def table */
	table.def td.footnote {
		padding-top: 0.6em;
	}
	table.def td.footnote::before {
		content: " ";
		display: block;
		height: 0.6em;
		width: 4em;
		border-top: thin solid;
	}

/** Data tables (and properly marked-up index tables) *************************/
	/*
		<table class="data"> highlights structural relationships in a table
		when correct markup is used (e.g. thead/tbody, th vs. td, scope attribute)

		Use class="complex data" for particularly complicated tables --
		(This will draw more lines: busier, but clearer.)

		Use class="long" on table cells with paragraph-like contents
		(This will adjust text alignment accordingly.)
		Alternately use class="longlastcol" on tables, to have the last column assume "long".
	*/

	table {
		word-wrap: normal;
		overflow-wrap: normal;
		hyphens: manual;
	}

	table.data,
	table.index {
		margin: 1em auto;
		border-collapse: collapse;
		border: hidden;
		width: 100%;
	}
	table.data caption,
	table.index caption {
		max-width: 50em;
		margin: 0 auto 1em;
	}

	table.data td,  table.data th,
	table.index td, table.index th {
		padding: 0.5em 1em;
		border-width: 1px;
		border-color: silver;
		border-color: var(--datacell-border);
		border-top-style: solid;
	}

	table.data thead td:empty {
		padding: 0;
		border: 0;
	}

	table.data  thead,
	table.index thead,
	table.data  tbody,
	table.index tbody {
		border-bottom: 2px solid;
	}

	table.data colgroup,
	table.index colgroup {
		border-left: 2px solid;
	}

	table.data  tbody th:first-child,
	table.index tbody th:first-child  {
		border-right: 2px solid;
		border-top: 1px solid silver;
		border-top: 1px solid var(--datacell-border);
		padding-right: 1em;
	}

	table.data th[colspan],
	table.data td[colspan] {
		text-align: center;
	}

	table.complex.data th,
	table.complex.data td {
		border: 1px solid silver;
		border: 1px solid var(--datacell-border);
		text-align: center;
	}

	table.data.longlastcol td:last-child,
	table.data td.long {
		vertical-align: baseline;
		text-align: left;
	}

	table.data img {
		vertical-align: middle;
	}


/*
Alternate table alignment rules

	table.data,
	table.index {
		text-align: center;
	}

	table.data  thead th[scope="row"],
	table.index thead th[scope="row"] {
		text-align: right;
	}

	table.data  tbody th:first-child,
	table.index tbody th:first-child  {
		text-align: right;
	}

Possible extra rowspan handling

	table.data  tbody th[rowspan]:not([rowspan='1']),
	table.index tbody th[rowspan]:not([rowspan='1']),
	table.data  tbody td[rowspan]:not([rowspan='1']),
	table.index tbody td[rowspan]:not([rowspan='1']) {
		border-left: 1px solid silver;
	}

	table.data  tbody th[rowspan]:first-child,
	table.index tbody th[rowspan]:first-child,
	table.data  tbody td[rowspan]:first-child,
	table.index tbody td[rowspan]:first-child{
		border-left: 0;
		border-right: 1px solid silver;
	}
*/

/******************************************************************************/
/*                                  Indices                                   */
/******************************************************************************/


/** Table of Contents *********************************************************/

	.toc a {
		/* More spacing; use padding to make it part of the click target. */
		padding: 0.1rem 1px 0;
		/* Larger, more consistently-sized click target */
		display: block;
		/* Switch to using border-bottom for underlines */
		text-decoration: none;
		border-bottom: 1px solid;
		/* Reverse color scheme */
		color: black;
		color: var(--toclink-text);
		border-color: #3980b5;
		border-color: var(--toclink-underline);
	}
	.toc a:visited {
		color: black;
		color: var(--toclink-visited-text);
		border-color: #054572;
		border-color: var(--toclink-visited-underline);
	}
	.toc a:focus,
	.toc a:hover {
		background: rgba(75%, 75%, 75%, .25);
		background: var(--a-hover-bg);
		border-bottom-width: 3px;
		margin-bottom: -2px;
	}
	.toc a:not(:focus):not(:hover) {
		/* Allow colors to cascade through from link styling */
		border-bottom-color: transparent;
	}

	.toc, .toc ol, .toc ul, .toc li {
		list-style: none; /* Numbers must be inlined into source */
		/* because generated content isn't search/selectable and markers can't do multilevel yet */
		margin:  0;
		padding: 0;
	}
	.toc {
		line-height: 1.1em;
	}

	/* ToC not indented until third level, but font style & margins show hierarchy */
	.toc > li			{ font-weight: bold;   }
	.toc > li li		 { font-weight: normal; }
	.toc > li li li	  { font-size:   95%;	}
	.toc > li li li li	{ font-size:   90%;	}
	.toc > li li li li li { font-size:   85%;	}

	/* @supports not (display:grid) { */
		.toc > li			{ margin: 1.5rem 0;	}
		.toc > li li		 { margin: 0.3rem 0;	}
		.toc > li li li	  { margin-left: 2rem;   }

		/* Section numbers in a column of their own */
		.toc .secno {
			float: left;
			width: 4rem;
			white-space: nowrap;
		}
		.toc > li li li li .secno { font-size: 85%; }
		.toc > li li li li li .secno { font-size: 100%; }

		.toc li {
			clear: both;
		}

		:not(li) > .toc			 { margin-left:  5rem; }
		.toc .secno				 { margin-left: -5rem; }
		.toc > li li li .secno	  { margin-left: -7rem; }
		.toc > li li li li .secno	{ margin-left: -9rem; }
		.toc > li li li li li .secno { margin-left: -11rem; }

		/* Tighten up indentation in narrow ToCs */
		@media (max-width: 30em) {
			:not(li) > .toc			 { margin-left:  4rem; }
			.toc .secno				 { margin-left: -4rem; }
			.toc > li li li			 { margin-left:  1rem; }
			.toc > li li li .secno	  { margin-left: -5rem; }
			.toc > li li li li .secno	{ margin-left: -6rem; }
			.toc > li li li li li .secno { margin-left: -7rem; }
		}
		/* Loosen it on wide screens */
		@media screen and (min-width: 78em) {
			body:not(.toc-inline) :not(li) > .toc			 { margin-left:  4rem; }
			body:not(.toc-inline) .toc .secno				 { margin-left: -4rem; }
			body:not(.toc-inline) .toc > li li li			 { margin-left:  1rem; }
			body:not(.toc-inline) .toc > li li li .secno	  { margin-left: -5rem; }
			body:not(.toc-inline) .toc > li li li li .secno	{ margin-left: -6rem; }
			body:not(.toc-inline) .toc > li li li li li .secno { margin-left: -7rem; }
	}
	/* } */

	@supports (display:grid) and (display:contents) {
		/* Use #toc over .toc to override non-@supports rules. */
		#toc {
			display: grid;
			align-content: start;
			grid-template-columns: auto 1fr;
			grid-column-gap: 1rem;
			column-gap: 1rem;
			grid-row-gap: .6rem;
			row-gap: .6rem;
		}
		#toc h2 {
			grid-column: 1 / -1;
			margin-bottom: 0;
		}
		#toc ol,
		#toc li,
		#toc a {
			display: contents;
			/* Switch <a> to subgrid when supported */
		}
		#toc span {
			margin: 0;
		}
		#toc > .toc > li > a > span {
			/* The spans of the top-level list,
			  comprising the first items of each top-level section. */
			margin-top: 1.1rem;
		}
		#toc#toc .secno { /* Ugh, need more specificity to override base.css */
			grid-column: 1;
			width: auto;
			margin-left: 0;
		}
		#toc .content {
			grid-column: 2;
			width: auto;
			margin-right: 1rem;
			border-bottom: 3px solid transparent;
			margin-bottom: -3px;
		}
		#toc .content:hover,
		#toc .content:focus {
			background: rgba(75%, 75%, 75%, .25);
			background: var(--a-hover-bg);
			border-bottom-color: #054572;
			border-bottom-color: var(--toclink-underline);
		}
		#toc li li li .content {
			margin-left: 1rem;
		}
		#toc li li li li .content {
			margin-left: 2rem;
		}
	}


/** Index *********************************************************************/

	/* Index Lists: Layout */
	ul.index	  { margin-left: 0; columns: 15em; text-indent: 1em hanging; }
	ul.index li	{ margin-left: 0; list-style: none; break-inside: avoid; }
	ul.index li li { margin-left: 1em; }
	ul.index dl	{ margin-top: 0; }
	ul.index dt	{ margin: .2em 0 .2em 20px;}
	ul.index dd	{ margin: .2em 0 .2em 40px;}
	/* Index Lists: Typography */
	ul.index ul,
	ul.index dl { font-size: smaller; }
	@media not print {
		ul.index li a + span {
			white-space: nowrap;
			color: transparent; }
		ul.index li a:hover + span,
		ul.index li a:focus + span {
			color: #707070;
			color: var(--indexinfo-text);
		}
	}

/** Index Tables *****************************************************/
	/* See also the data table styling section, which this effectively subclasses */

	table.index {
		font-size: small;
		border-collapse: collapse;
		border-spacing: 0;
		text-align: left;
		margin: 1em 0;
	}

	table.index td,
	table.index th {
		padding: 0.4em;
	}

	table.index tr:hover td:not([rowspan]),
	table.index tr:hover th:not([rowspan]) {
		color: black;
		color: var(--indextable-hover-text);
		background: #f7f8f9;
		background: var(--indextable-hover-bg);
	}

	/* The link in the first column in the property table (formerly a TD) */
	table.index th:first-child a {
		font-weight: bold;
	}

/** Outdated warning **********************************************************/

.outdated-spec {
	color: black;
	color: var(--outdatedspec-text);
	background-color: rgba(0,0,0,0.5);
	background-color: var(--outdatedspec-bg);
}

.outdated-warning {
	position: fixed;
	bottom: 50%;
	left: 0;
	right: 0;
	margin: 0 auto;
	width: 50%;
	background: maroon;
	background: var(--outdated-bg);
	color: white;
	color: var(--outdated-text);
	border-radius: 1em;
	box-shadow: 0 0 1em red;
	box-shadow: 0 0 1em var(--outdated-shadow);
	padding: 2em;
	text-align: center;
	z-index: 2;
}

.outdated-warning a {
	color: currentcolor;
	background: transparent;
}

.edited-rec-warning {
	background: darkorange;
	background: var(--editedrec-bg);
	box-shadow: 0 0 1em;
}

.outdated-warning button {
	color: var(--outdated-text);
	border-radius: 1em;
	box-shadow: 0 0 1em red;
	box-shadow: 0 0 1em var(--outdated-shadow);
	padding: 2em;
	text-align: center;
	z-index: 2;
}

.outdated-warning a {
	color: currentcolor;
	background: transparent;
}

.edited-rec-warning {
	background: darkorange;
	background: var(--editedrec-bg);
	box-shadow: 0 0 1em;
}

.outdated-warning button {
	position: absolute;
	top: 0;
	right:0;
	margin: 0;
	border: 0;
	padding: 0.25em 0.5em;
	background: transparent;
	color: white;
	color: var(--outdated-text);
	font:1em sans-serif;
	text-align:center;
}

.outdated-warning span {
	display: block;
}

.outdated-collapsed {
	bottom: 0;
	border-radius: 0;
	width: 100%;
	padding: 0;
}

/******************************************************************************/
/*                                    Print                                   */
/******************************************************************************/

	@media print {
		/* Pages have their own margins. */
		html {
			margin: 0;
		}
		/* Serif for print. */
		body {
			font-family: serif;
		}

		.outdated-warning {
			position: absolute;
			border-style: solid;
			border-color: red;
		}

		.outdated-warning input {
			display: none;
		}
	}
	@page {
		margin: 1.5cm 1.1cm;
	}



/******************************************************************************/
/*                             Overflow Control                               */
/******************************************************************************/

	.figure .caption, .sidefigure .caption, figcaption {
		/* in case figure is overlarge, limit caption to 50em */
		max-width: 50rem;
		margin-left: auto;
		margin-right: auto;
	}
	.overlarge {
		/* Magic to create good item positioning:
		  "content column" is 50ems wide at max; less on smaller screens.
		  Extra space (after ToC + content) is empty on the right.

		  1. When item < content column, centers item in column.
		  2. When content < item < available, left-aligns.
		  3. When item > available, fills available + scroll bar.
		*/
		display: grid;
		grid-template-columns: minmax(0, 50em);
	}
	.overlarge > table {
		/* limit preferred width of table */
		max-width: 50em;
		margin-left: auto;
		margin-right: auto;
	}

	@media (min-width: 55em) {
		.overlarge {
			margin-right: calc(13px + 26.5rem - 50vw);
			max-width: none;
		}
	}
	@media screen and (min-width: 78em) {
		body:not(.toc-inline) .overlarge {
			/* 30.5em body padding 50em content area */
			margin-right: calc(40em - 50vw) !important;
		}
	}
	@media screen and (min-width: 90em) {
		body:not(.toc-inline) .overlarge {
			/* 4em html margin 30.5em body padding 50em content area */
			margin-right: calc(84.5em - 100vw) !important;
		}
	}

	@media not print {
		.overlarge {
			overflow-x: auto;
			/* See Lea Verou's explanation background-attachment:
			* http://lea.verou.me/2012/04/background-attachment-local/
			*
			background: top left  / 4em 100% linear-gradient(to right,  #ffffff, rgba(255, 255, 255, 0)) local,
						top right / 4em 100% linear-gradient(to left, #ffffff, rgba(255, 255, 255, 0)) local,
						top left  / 1em 100% linear-gradient(to right,  #c3c3c5, rgba(195, 195, 197, 0)) scroll,
						top right / 1em 100% linear-gradient(to left, #c3c3c5, rgba(195, 195, 197, 0)) scroll,
						white;
			background-repeat: no-repeat;
			*/
		}
	}
</style>
  <meta content="Bikeshed version 0ef803fdf, updated Tue Jan 6 11:59:39 2026 -0800" name="generator">
  <link href="TBD" rel="canonical">
  <meta content="40daf9c29bcc27b8afda16b751411f32b6b5e4b0" name="revision">
  <meta content="dark light" name="color-scheme">
<style>/* Boilerplate: style-autolinks */
.css.css, .property.property, .descriptor.descriptor {
    color: var(--a-normal-text);
    font-size: inherit;
    font-family: inherit;
}
.css::before, .property::before, .descriptor::before {
    content: "‘";
}
.css::after, .property::after, .descriptor::after {
    content: "’";
}
.property, .descriptor {
    /* Don't wrap property and descriptor names */
    white-space: nowrap;
}
.type { /* CSS value <type> */
    font-style: italic;
}
pre .property::before, pre .property::after {
    content: "";
}
[data-link-type="property"]::before,
[data-link-type="propdesc"]::before,
[data-link-type="descriptor"]::before,
[data-link-type="value"]::before,
[data-link-type="function"]::before,
[data-link-type="at-rule"]::before,
[data-link-type="selector"]::before,
[data-link-type="maybe"]::before {
    content: "‘";
}
[data-link-type="property"]::after,
[data-link-type="propdesc"]::after,
[data-link-type="descriptor"]::after,
[data-link-type="value"]::after,
[data-link-type="function"]::after,
[data-link-type="at-rule"]::after,
[data-link-type="selector"]::after,
[data-link-type="maybe"]::after {
    content: "’";
}

[data-link-type].production::before,
[data-link-type].production::after,
.prod [data-link-type]::before,
.prod [data-link-type]::after {
    content: "";
}

[data-link-type=element],
[data-link-type=element-attr] {
    font-family: Menlo, Consolas, "DejaVu Sans Mono", monospace;
    font-size: .9em;
}
[data-link-type=element]::before { content: "<" }
[data-link-type=element]::after  { content: ">" }

[data-link-type=biblio] {
    white-space: pre;
}

@media (prefers-color-scheme: dark) {
    :root {
        --selflink-text: black;
        --selflink-bg: silver;
        --selflink-hover-text: white;
    }
}
</style>
<style>/* Boilerplate: style-colors */
/* Any --*-text not paired with a --*-bg is assumed to have a transparent bg */
:root {
    color-scheme: light dark;

    --text: black;
    --bg: white;

    --unofficial-watermark: url(https://www.w3.org/StyleSheets/TR/2016/logos/UD-watermark);

    --logo-bg: #1a5e9a;
    --logo-active-bg: #c00;
    --logo-text: white;

    --tocnav-normal-text: #707070;
    --tocnav-normal-bg: var(--bg);
    --tocnav-hover-text: var(--tocnav-normal-text);
    --tocnav-hover-bg: #f8f8f8;
    --tocnav-active-text: #c00;
    --tocnav-active-bg: var(--tocnav-normal-bg);

    --tocsidebar-text: var(--text);
    --tocsidebar-bg: #f7f8f9;
    --tocsidebar-shadow: rgba(0,0,0,.1);
    --tocsidebar-heading-text: hsla(203,20%,40%,.7);

    --toclink-text: var(--text);
    --toclink-underline: #3980b5;
    --toclink-visited-text: var(--toclink-text);
    --toclink-visited-underline: #054572;

    --heading-text: #005a9c;

    --hr-text: var(--text);

    --algo-border: #def;

    --del-text: red;
    --del-bg: transparent;
    --ins-text: #080;
    --ins-bg: transparent;

    --a-normal-text: #034575;
    --a-normal-underline: #bbb;
    --a-visited-text: var(--a-normal-text);
    --a-visited-underline: #707070;
    --a-hover-bg: rgba(75%, 75%, 75%, .25);
    --a-active-text: #c00;
    --a-active-underline: #c00;

    --blockquote-border: silver;
    --blockquote-bg: transparent;
    --blockquote-text: currentcolor;

    --issue-border: #e05252;
    --issue-bg: #fbe9e9;
    --issue-text: var(--text);
    --issueheading-text: #831616;

    --example-border: #e0cb52;
    --example-bg: #fcfaee;
    --example-text: var(--text);
    --exampleheading-text: #574b0f;

    --note-border: #52e052;
    --note-bg: #e9fbe9;
    --note-text: var(--text);
    --noteheading-text: hsl(120, 70%, 30%);
    --notesummary-underline: silver;

    --assertion-border: #aaa;
    --assertion-bg: #eee;
    --assertion-text: black;

    --advisement-border: orange;
    --advisement-bg: #fec;
    --advisement-text: var(--text);
    --advisementheading-text: #b35f00;

    --warning-border: red;
    --warning-bg: hsla(40,100%,50%,0.95);
    --warning-text: var(--text);

    --amendment-border: #330099;
    --amendment-bg: #F5F0FF;
    --amendment-text: var(--text);
    --amendmentheading-text: #220066;

    --def-border: #8ccbf2;
    --def-bg: #def;
    --def-text: var(--text);
    --defrow-border: #bbd7e9;

    --datacell-border: silver;

    --indexinfo-text: #707070;

    --indextable-hover-text: black;
    --indextable-hover-bg: #f7f8f9;

    --outdatedspec-bg: rgba(0, 0, 0, .5);
    --outdatedspec-text: black;
    --outdated-bg: maroon;
    --outdated-text: white;
    --outdated-shadow: red;

    --editedrec-bg: darkorange;
}

@media (prefers-color-scheme: dark) {
    :root {
        --text: #ddd;
        --bg: black;

        --unofficial-watermark: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='400' height='400'%3E%3Cg fill='%23100808' transform='translate(200 200) rotate(-45) translate(-200 -200)' stroke='%23100808' stroke-width='3'%3E%3Ctext x='50%25' y='220' style='font: bold 70px sans-serif; text-anchor: middle; letter-spacing: 6px;'%3EUNOFFICIAL%3C/text%3E%3Ctext x='50%25' y='305' style='font: bold 70px sans-serif; text-anchor: middle; letter-spacing: 6px;'%3EDRAFT%3C/text%3E%3C/g%3E%3C/svg%3E");

        --logo-bg: #1a5e9a;
        --logo-active-bg: #c00;
        --logo-text: white;

        --tocnav-normal-text: #999;
        --tocnav-normal-bg: var(--bg);
        --tocnav-hover-text: var(--tocnav-normal-text);
        --tocnav-hover-bg: #080808;
        --tocnav-active-text: #f44;
        --tocnav-active-bg: var(--tocnav-normal-bg);

        --tocsidebar-text: var(--text);
        --tocsidebar-bg: #080808;
        --tocsidebar-shadow: rgba(255,255,255,.1);
        --tocsidebar-heading-text: hsla(203,20%,40%,.7);

        --toclink-text: var(--text);
        --toclink-underline: #6af;
        --toclink-visited-text: var(--toclink-text);
        --toclink-visited-underline: #054572;

        --heading-text: #8af;

        --hr-text: var(--text);

        --algo-border: #456;

        --del-text: #f44;
        --del-bg: transparent;
        --ins-text: #4a4;
        --ins-bg: transparent;

        --a-normal-text: #6af;
        --a-normal-underline: #555;
        --a-visited-text: var(--a-normal-text);
        --a-visited-underline: var(--a-normal-underline);
        --a-hover-bg: rgba(25%, 25%, 25%, .2);
        --a-active-text: #f44;
        --a-active-underline: var(--a-active-text);

        --borderedblock-bg: rgba(255, 255, 255, .05);

        --blockquote-border: silver;
        --blockquote-bg: var(--borderedblock-bg);
        --blockquote-text: currentcolor;

        --issue-border: #e05252;
        --issue-bg: var(--borderedblock-bg);
        --issue-text: var(--text);
        --issueheading-text: hsl(0deg, 70%, 70%);

        --example-border: hsl(50deg, 90%, 60%);
        --example-bg: var(--borderedblock-bg);
        --example-text: var(--text);
        --exampleheading-text: hsl(50deg, 70%, 70%);

        --note-border: hsl(120deg, 100%, 35%);
        --note-bg: var(--borderedblock-bg);
        --note-text: var(--text);
        --noteheading-text: hsl(120, 70%, 70%);
        --notesummary-underline: silver;

        --assertion-border: #444;
        --assertion-bg: var(--borderedblock-bg);
        --assertion-text: var(--text);

        --advisement-border: orange;
        --advisement-bg: #222218;
        --advisement-text: var(--text);
        --advisementheading-text: #f84;

        --warning-border: red;
        --warning-bg: hsla(40,100%,20%,0.95);
        --warning-text: var(--text);

        --amendment-border: #330099;
        --amendment-bg: #080010;
        --amendment-text: var(--text);
        --amendmentheading-text: #cc00ff;

        --def-border: #8ccbf2;
        --def-bg: #080818;
        --def-text: var(--text);
        --defrow-border: #136;

        --datacell-border: silver;

        --indexinfo-text: #aaa;

        --indextable-hover-text: var(--text);
        --indextable-hover-bg: #181818;

        --outdatedspec-bg: rgba(255, 255, 255, .5);
        --outdatedspec-text: black;
        --outdated-bg: maroon;
        --outdated-text: white;
        --outdated-shadow: red;

        --editedrec-bg: darkorange;
    }
    /* In case a transparent-bg image doesn't expect to be on a dark bg,
       which is quite common in practice... */
    img { background: white; }
}
</style>
<style>/* Boilerplate: style-counters */
body {
    counter-reset: example figure issue;
}
.issue {
    counter-increment: issue;
}
.issue:not(.no-marker)::before {
    content: "Issue " counter(issue);
}

.example {
    counter-increment: example;
}
.example:not(.no-marker)::before {
    content: "Example " counter(example);
}
.invalid.example:not(.no-marker)::before,
.illegal.example:not(.no-marker)::before {
    content: "Invalid Example " counter(example);
}

figcaption {
    counter-increment: figure;
}
figcaption:not(.no-marker)::before {
    content: "Figure " counter(figure) " ";
}
</style>
<style>/* Boilerplate: style-issues */
a[href].issue-return {
    float: right;
    float: inline-end;
    color: var(--issueheading-text);
    font-weight: bold;
    text-decoration: none;
}
</style>
<style>/* Boilerplate: style-md-lists */
/* This is a weird hack for me not yet following the commonmark spec
   regarding paragraph and lists. */
[data-md] > :first-child {
    margin-top: 0;
}
[data-md] > :last-child {
    margin-bottom: 0;
}
</style>
<style>/* Boilerplate: style-selflinks */
:root {
    --selflink-text: white;
    --selflink-bg: gray;
    --selflink-hover-text: black;
}
.heading, .issue, .note, .example, li, dt {
    position: relative;
}
a.self-link {
    position: absolute;
    top: 0;
    left: calc(-1 * (3.5rem - 26px));
    width: calc(3.5rem - 26px);
    height: 2em;
    text-align: center;
    border: none;
    transition: opacity .2s;
    opacity: .5;
}
a.self-link:hover {
    opacity: 1;
}
.heading > a.self-link {
    font-size: 83%;
}
.example > a.self-link,
.note > a.self-link,
.issue > a.self-link {
    /* These blocks are overflow:auto, so positioning outside
       doesn't work. */
    left: auto;
    right: 0;
}
li > a.self-link {
    left: calc(-1 * (3.5rem - 26px) - 2em);
}
dfn > a.self-link {
    top: auto;
    left: auto;
    opacity: 0;
    width: 1.5em;
    height: 1.5em;
    background: var(--selflink-bg);
    color: var(--selflink-text);
    font-style: normal;
    transition: opacity .2s, background-color .2s, color .2s;
}
dfn:hover > a.self-link {
    opacity: 1;
}
dfn > a.self-link:hover {
    color: var(--selflink-hover-text);
}

a.self-link::before            { content: "¶"; }
.heading > a.self-link::before { content: "§"; }
dfn > a.self-link::before      { content: "#"; }
</style>
 <body class="h-entry">
  <div class="head">
   <p data-fill-with="logo"></p>
   <h1 class="no-ref p-name" id="title">GDI Harmonised Minimal Data Model</h1>
   <h2 class="heading no-num no-ref no-toc settled" id="profile-and-date"><span class="content">Living Document,
    <time class="dt-updated" datetime="2026-01-08">8 January 2026</time></span></h2>
   <div data-fill-with="spec-metadata">
    <dl>
     <dt>This version:
     <dd><a class="u-url" href="TBD">TBD</a>
     <dt>Issue Tracking:
     <dd><a href="https://github.com/Health-RI/HarmonisedMinimalDatamodel/issues/">GitHub</a>
     <dt class="editor">Editors:
     <dd class="editor h-card p-author vcard"><span class="fn p-name">Ana Konrad</span>
     <dd class="editor h-card p-author vcard"><span class="fn p-name">Hannah Neikes</span>
     <dd class="editor h-card p-author vcard"><span class="fn p-name">Jeroen Beliën</span>
     <dd class="editor h-card p-author vcard"><span class="fn p-name">Joeri van der Velde</span>
     <dd class="editor h-card p-author vcard"><span class="fn p-name">Abhishek Nayak</span>
     <dd class="editor h-card p-author vcard"><span class="fn p-name">Aedin Culhane</span>
    </dl>
   </div>
   <div data-fill-with="warning">
    <details class="annoying-warning" open>
     <summary>Not Ready For Implementation</summary>
     <p>
		This spec is not yet ready for implementation.
		It exists in this repository to record the ideas and promote discussion.

	</p>
     <p>
		Before attempting to implement this spec,
		please contact the editors.
</p>
    </details>
   </div>
   <p class="copyright" data-fill-with="copyright"><a href="http://creativecommons.org/publicdomain/zero/1.0/" rel="license"><img alt="CC0" height="15" src="https://licensebuttons.net/p/zero/1.0/80x15.png" width="80"></a>
To the extent possible under law, the editors have waived all copyright
and related or neighboring rights to this work.
In addition, as of 8 January 2026,
the editors have made this specification available under the
<a href="https://www.openwebfoundation.org/the-agreements/the-owf-1-0-agreements-granted-claims/owfa-1-0" rel="license">Open Web Foundation Agreement Version 1.0</a>,
which is available at https://www.openwebfoundation.org/the-agreements/the-owf-1-0-agreements-granted-claims/owfa-1-0.
Parts of this work may be from another specification document.  If so, those parts are instead covered by the license of that specification document.
</p>
   <hr title="Separator for header">
  </div>
  <div class="p-summary" data-fill-with="abstract">
   <h2 class="heading no-num no-ref no-toc settled" id="abstract"><span class="content">Abstract</span></h2>
   <p><img alt="GDI-logo" src="src/new/images/gdi-logo.svg" title="GDI-logo"> <br> Some context about the GDI HMD.</p>
  </div>
  <div data-fill-with="at-risk"></div>
  <nav data-fill-with="table-of-contents" id="toc">
   <h2 class="no-num no-ref no-toc" id="contents">Table of Contents</h2>
   <ol class="toc">
    <li>
     <a href="#introduction"><span class="secno">1</span> <span class="content">Introduction</span></a>
     <ol class="toc">
      <li><a href="#goals-and-scope-of-the-harmonized-minimal-data-model"><span class="secno">1.1</span> <span class="content">Goals and Scope of the Harmonized Minimal Data Model</span></a>
      <li><a href="#overview-and-diagram"><span class="secno">1.2</span> <span class="content">Overview and Diagram</span></a>
     </ol>
    <li>
     <a href="#main-classes"><span class="secno">2</span> <span class="content">Main Classes</span></a>
     <ol class="toc">
      <li><a href="#subject"><span class="secno">2.1</span> <span class="content">Subject</span></a>
      <li><a href="#diagnosis"><span class="secno">2.2</span> <span class="content">Diagnosis</span></a>
      <li>
       <a href="#sample"><span class="secno">2.3</span> <span class="content">Sample</span></a>
       <ol class="toc">
        <li><a href="#tnm"><span class="secno">2.3.1</span> <span class="content">TNM</span></a>
        <li><a href="#sample-relation"><span class="secno">2.3.2</span> <span class="content">Sample Relation</span></a>
       </ol>
      <li>
       <a href="#sequence"><span class="secno">2.4</span> <span class="content">Sequence</span></a>
       <ol class="toc">
        <li><a href="#digital-resource"><span class="secno">2.4.1</span> <span class="content">Digital Resource</span></a>
        <li><a href="#variant"><span class="secno">2.4.2</span> <span class="content">Variant</span></a>
        <li><a href="#target-gene"><span class="secno">2.4.3</span> <span class="content">Target Gene</span></a>
       </ol>
      <li><a href="#treatment"><span class="secno">2.5</span> <span class="content">Treatment</span></a>
      <li><a href="#biomarker"><span class="secno">2.6</span> <span class="content">Biomarker</span></a>
      <li><a href="#environmental-exposure"><span class="secno">2.7</span> <span class="content">Environmental Exposure</span></a>
     </ol>
    <li>
     <a href="#matches"><span class="secno">3</span> <span class="content">Matches</span></a>
     <ol class="toc">
      <li><a href="#how-to-use-matches"><span class="secno">3.1</span> <span class="content">How To Use Matches</span></a>
      <li><a href="#using-sssom"><span class="secno">3.2</span> <span class="content">Using SSSOM</span></a>
     </ol>
    <li><a href="#conformance"><span class="secno"></span> <span class="content">
    Conformance</span></a>
    <li>
     <a href="#references"><span class="secno"></span> <span class="content">References</span></a>
     <ol class="toc">
      <li><a href="#normative"><span class="secno"></span> <span class="content">Normative References</span></a>
     </ol>
   </ol>
  </nav>
  <main>
   <h2 class="heading settled" data-level="1" id="introduction"><span class="secno">1. </span><span class="content">Introduction</span><a class="self-link" href="#introduction"></a></h2>
   
Some info about GDI here.


   <h3 class="heading settled" data-level="1.1" id="goals-and-scope-of-the-harmonized-minimal-data-model"><span class="secno">1.1. </span><span class="content">Goals and Scope of the Harmonized Minimal Data Model</span><a class="self-link" href="#goals-and-scope-of-the-harmonized-minimal-data-model"></a></h3>
   <p>The <strong>Harmonized Minimal Data Model (HMDM)</strong> has multiple purposes, some of these are:</p>
   <ol>
    <li data-md>
     <p>Each dataset that is submitted to the catalogue should adhere (to a certain extend, still to be decided upon) to this model</p>
    <li data-md>
     <p>Within the catalogue based on our model search options can be made available (filters/facets)</p>
    <li data-md>
     <p>The HMDM can/needs to be implemented in the Beacons (and/or other discovery tools).</p>
   </ol>
   <p>Next to the HMDM we provided a <strong>minimal metadata for submission schema</strong>.</p>
   <p>Version 1 of metadata for submission model was released and can be seen <a href="https://drive.google.com/drive/u/0/folders/1oojXociy39DD_VzUZHfx1n4queVJD1RE">here</a>. This version was based on the version of HealthDCAT-AP from December 2024. Since the work on HealthDCAT-AP is still ongoing, there have been some changes since that time. This means our metadata for submission is not compliant to the <a href="https://healthdcat-ap.github.io/">current version of HealthDCAT-AP</a> anymore.
We will update the metadata for submission to version 1.1, which will be, according to HealthDCAT-AP, provided in 3 variants, based on the access level of the dataset: <em>Open</em>, <em>Sensitive</em> and <em>Protected</em>, which differ in cardinalities of the properties. The metadata for submission model will include all the current mandatory fields from those HealthDCAT-AP vairants, as well as a custom class created specifically for the submission of metadata in GDI, called HMD Submission. 
The name spaces for HMD Submission class and its properties were made available through <a href="https://github.com/fairgenomes/fairgenomes-semantic-model">FAIR Genomes repository</a>.
There are also some mandatory fields that were already included in the GDI MS8 deliverable, which will stay mandatory for backwards compatibility. This update will be very minimal, so only extending the current model with new mandatory fields.</p>
   <h3 class="heading settled" data-level="1.2" id="overview-and-diagram"><span class="secno">1.2. </span><span class="content">Overview and Diagram</span><a class="self-link" href="#overview-and-diagram"></a></h3>
   <h2 class="heading settled" data-level="2" id="main-classes"><span class="secno">2. </span><span class="content">Main Classes</span><a class="self-link" href="#main-classes"></a></h2>
   
In the Harmonized Minimal Dataset, we defined 12 different classes.





   <h3 class="heading settled" data-level="2.1" id="subject"><span class="secno">2.1. </span><span class="content">Subject</span><a class="self-link" href="#subject"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Birth Date
      <td>The calendar date on which a person was born.
      <td>Complete date, without time, following the ISO 8601. If only year or year-month is available, use that. xsd:date or xsd:gYearMonth or xsd:gYear
      <td>1..1
     <tr>
      <td>Administrative Gender
      <td>The gender of a person used for administrative purposes.
      <td><a href="https://build.fhir.org/valueset-administrative-gender.html">HL7 Administrative Gender</a>
      <td>1..1
     <tr>
      <td>biological sex at birth
      <td>The sex of a person at birth as molecular proven.
      <td><a href="https://www.hl7.org/fhir/us/core/ValueSet-birthsex.html">HL7 ValueSet: Birth Sex</a>
      <td>0..1
     <tr>
      <td>Date_of_Last_Follow_up
      <td>Date of last follow-up, partial date with month and year.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
     <tr>
      <td>Country of Origin
      <td>A person’s descent or lineage, from a person or from a population.
      <td>2- or 3-lettercode from <a href="https://www.iso.org/obp/ui/#iso:std:iso:3166:-1:ed-4:v1:en">ISO 3166-1</a> if only a country code is provided. If a country-subdivision then a value from the <a href="https://www.iso.org/obp/ui/#iso:std:iso:3166:-2:ed-4:v1:en">ISO 3166-2</a>
      <td>0..1
     <tr>
      <td>Subject ID
      <td>A sequence of characters used to identify, name, or characterize a trial or study subject.
      <td>String
      <td>1..1
   </table>
   <h3 class="heading settled" data-level="2.2" id="diagnosis"><span class="secno">2.2. </span><span class="content">Diagnosis</span><a class="self-link" href="#diagnosis"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Date of Diagnosis
      <td>Date at which diagnosis was made.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
     <tr>
      <td>Diagnosis
      <td>The investigation, analysis and recognition of the presence and nature of disease, condition, or injury from expressed signs and symptoms; also, the scientific determination of any kind; the concise results of such an investigation.
      <td>Children of SNOMED-CT <a href="http://purl.bioontology.org/ontology/SNOMEDCT/64572001">Disease (Disorder)</a>
      <td>0..n
     <tr>
      <td>Provisional diagnosis / clinical diagnosis
      <td>An initial diagnosis that is subject to change as new information becomes available.
      <td>Children of SNOMED-CT <a href="http://purl.bioontology.org/ontology/SNOMEDCT/64572001">Disease (Disorder)</a>
      <td>0..n
   </table>
   <h3 class="heading settled" data-level="2.3" id="sample"><span class="secno">2.3. </span><span class="content">Sample</span><a class="self-link" href="#sample"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Anatomical sample location
      <td>Anatomic site from which the sample was taken.
      <td>ICD-11 <a href="https://icd.who.int/browse/2024-01/mms/en#1154280071">Anatomy and topography</a>
      <td>1..1
     <tr>
      <td>Pathological state
      <td>The pathological condition of the sample.
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C162623">Tissue Normal</a>, Germline Normal, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C8509">Primary Tumor</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C3261">Tumor Metastasis</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C4798">Recurrent Tumor</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C172259">Organoid</a>, <a href="https://w3id.org/fair-genomes/resource/FG_0000124">Tumoroid</a>
      <td>1..1
     <tr>
      <td>Date
      <td>Defines the date of sampling.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
     <tr>
      <td>ID
      <td>Unique identifier for a collected specimen assigned by data provider.
      <td>String
      <td>0..1
     <tr>
      <td>Organism
      <td>A living entity.
      <td>Children of <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C14250">NCIT Organism</a>
      <td>1..1
     <tr>
      <td>Biospecimen_Type
      <td>The type of a material sample taken from a biological entity for testing, diagnostic, propagation, treatment or research purposes. This includes particular types of cellular molecules, cells, tissues, organs, body fluids, embryos, and body excretory substances.
      <td>Value from type of Sample table in <a href="https://cdn.ymaws.com/www.isber.org/resource/resmgr/isber_2019/pdf/standard_preanalytical_code_.pdf">SPREC Codes v3.0 </a>:
            <br>ASC - Ascites fluid
            <br> AMN - Amniotic fluid
            <br> BAL - Bronchoalveolar lavage
            <br> BLD - Blood (whole)
            <br> BMA - Bone marrow aspirate
            <br> BMK - Breast milk
            <br> BUC - Buccal cells
            <br> BUF - Unficolled buffy coat, viable
            <br> CEL - Ficoll mononuclear cells, viable
            <br> CEN - Fresh cells from non blood specimen type
            <br> CLN - Cells from nonblood specimen type (e.g., disrupted tissue), viable
            <br> CRD - Cord blood
            <br> CSF - Cerebrospinal fluid
            <br> NAS - Nasal washing
            <br> PEL - Ficoll mononuclear cells, nonviable
            <br> PEN - Cells from nonblood specimen type (e.g., disrupted tissue), nonviable
            <br> PFL - Pleural fluid
            <br> PL1 - Plasma, single spun
            <br> PL2 - Plasma, double spun
            <br> SAL - Saliva
            <br> SEM - Semen
            <br> SER - Serum
            <br> SPT - Sputum
            <br> STL - Stool
            <br> SYN - Synovial fluid
            <br> TER - Tears
            <br> U24 - 24-h urine
            <br> URN - Urine
            <br> ZZZ Other 
      <td>1..1
     <tr>
      <td>Extraction_Technique
      <td>The technique of extraction of the sample.
      <td>Values within <a href="http://purl.bioontology.org/ontology/MESH/D013048">MESH Specimen Handling</a>
      <td>0..1
     <tr>
      <td>Storage_Conditions
      <td>Storage conditions of the sample.
      <td><a href="https://github.com/fairgenomes/fairgenomes-semantic-model/blob/main/lookups/StorageConditions.txt"> FAIR Genomes Storage Conditions</a>
      <td>0..n
     <tr>
      <td>Assayed_Biological_Macromolecule
      <td>Macromolecule derived from the sample.
      <td>Children of <a href="http://www.ebi.ac.uk/efo/EFO_0004446">EFO biological macromolecule</a>
      <td>0..1
     <tr>
      <td>Sampling_Intent
      <td>Describes the purpose for taking the sample.
      <td>String
      <td>0..n
   </table>
   <h4 class="heading settled" data-level="2.3.1" id="tnm"><span class="secno">2.3.1. </span><span class="content">TNM</span><a class="self-link" href="#tnm"></a></h4>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Tumor_size
      <td>Indicates the size of the primary tumor.
      <td>Double
      <td>0..1
    <tbody>
     <tr>
      <td>Tumor_size_unit
      <td>Indicates the unit in which tumor size is stated.
      <td><a href="https://ucum.org/">UCUM</a>
      <td>0..1
    <tbody>
     <tr>
      <td>Lymph_Node_Status
      <td>Indicates lymph node involvement according to the international TNM classification for solid tumors.
      <td>Children of <a href="http://purl.bioontology.org/ontology/SNOMEDCT/1279504008">SCTID: 1279504008</a>  | American Joint Committee on Cancer ycN category allowable value (qualifier value)
      <td>0..1
    <tbody>
     <tr>
      <td>Metastases
      <td>Indicates presence or absence of metastasis according to the international TNM classification for solid tumors.
      <td>Children of <a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222587001">SCTID: 1222587001 </a> | American Joint Committee on Cancer pathological M category allowable value (qualifier value), <br> or <br> Children of <a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222591006">SCTID: 1222591006 </a> | American Joint Committee on Cancer clinical M category allowable value (qualifier value), <br> or HL7 NULL flavor in case not determined.
      <td>0..1
    <tbody>
     <tr>
      <td>Version
      <td>Version of the TNM classification.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Stage
      <td>The extent of a cancer in the body. Staging is usually based on the size of the tumor, whether lymph nodes contain cancer, and whether the cancer has spread from the original site to other parts of the body.
      <td>Children of
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222592004">SCTID: 1222592004</a> |  American Joint Committee on Cancer clinical stage group allowable value (qualifier value), <br> or <br>
            <a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222593009">SCTID: 1222593009</a> | American Joint Committee on Cancer pathological stage group allowable value (qualifier value),<br> or
            <br> <a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222594003">SCTID: 1222594003</a> |  American Joint Committee on Cancer yp stage group allowable value (qualifier value),<br> or
            <br> UICC values (to be added to SNOMED in 2025) 
      <td>0..1
    <tbody>
     <tr>
      <td>Pathology_Clinical
      <td>Indication whether TNM classification is based on clinical or pathological evaluation. If pathological is available, this always goes before clinical.
      <td>Clinical or pathological
      <td>1..1
    <tbody>
     <tr>
      <td>Date of evaluation
      <td>Date of the clinical/pathological evaluation.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
   </table>
   <h4 class="heading settled" data-level="2.3.2" id="sample-relation"><span class="secno">2.3.2. </span><span class="content">Sample Relation</span><a class="self-link" href="#sample-relation"></a></h4>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Related_Sample
      <td>Points to a related sample.
      <td>Sample ID
      <td>1..n
    <tbody>
     <tr>
      <td>Relation_Type
      <td>Describes the relationship type between the two connected samples.
      <td>List to be defined/extended:
            <br>New sample
            <br>Derived
            <br>Aliquot
            <br>Control (experiment)
            <br>Disease versus Control
            <br>Matched
            <br>Longitudinal
            <br>Paired
            <br>Familial
            <br>Spatial
      <td>1..1
    <tbody>
     <tr>
      <td>Relation_Description
      <td>Free text describing the Relation Type of two samples.
      <td>String
      <td>0..1
   </table>
   <h3 class="heading settled" data-level="2.4" id="sequence"><span class="secno">2.4. </span><span class="content">Sequence</span><a class="self-link" href="#sequence"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Target
      <td>Identification of the sequenced target.
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C101294">Whole Genome Sequencing (WGS)</a> <br> <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C101295">Whole Exome Sequencing (WES)</a> <br> <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C198412">Multi-gene panel sequencing</a> <br>array <br> OTH 
      <td>1..1
    <tbody>
     <tr>
      <td>Sequencing_Date
      <td>Defines the date of sequencing.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
    <tbody>
     <tr>
      <td>ISO 15189 accredited
      <td>Indication whether the laboratory is accredited according to ISO 15189 (clinical).
      <td>Boolean
      <td>0..1
    <tbody>
     <tr>
      <td>ISO 17025 accredited
      <td>Indication whether the laboratory is accredited according to ISO 17025 (research).
      <td>Boolean
      <td>0..1
    <tbody>
     <tr>
      <td> <b>WIP</b>  <br> Wet lab protocol
      <td>
      <td>String or <a href="https://www.protocols.io">https://www.protocols.io</a> URL
      <td>
    <tbody>
     <tr>
      <td>Participated in proficiency testing
      <td>Indication whether the laboratory had participated in any proficiency testing, such as interlaboratory comparison.
      <td>Boolean
      <td>0..1
    <tbody>
     <tr>
      <td>IVDR_passed
      <td>Indicate whether the methodology (including chemistry and sequencing standards) used for sequencing follows the In vitro diagnostic medical devices (IVDR) regulation passed by the EU in April 2017.
      <td>Boolean
      <td>0..1
    <tbody>
     <tr>
      <td>Sequencing Platform
      <td>The used sequencing platform (i.e. brand, name of a company that produces sequencer equipment).
      <td><a href="https://github.com/fairgenomes/fairgenomes-semantic-model/blob/main/lookups/SequencingInstrumentModels.txt">FAIR Genomes</a> or <a href="http://purl.obolibrary.org/obo/OBI_0400103">EFO</a> list
      <td>1..n
    <tbody>
     <tr>
      <td>Average depth of coverage
      <td>Mean coverage for whole genome sequencing, or mean target coverage for whole exome and targeted sequencing (eg 60x, average number of times each target base has been ‘read’ by sequencer).
      <td>Integer
      <td>1..1
    <tbody>
     <tr>
      <td>Breadth of coverage
      <td>Breadth of coverage (or evenness) is the proportion or percentage of a reads that has been sequenced at a the provided average depth of coverage. (E.g. if for Average depth of coverage, the value is 60, and the Evenness is 50% at 60x, the value here will be 50).
      <td>Integer
      <td>0..1
    <tbody>
     <tr>
      <td>Additional NGS quality control metrics
      <td>Statement of any additional NGS quality control metrics.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Initial_input_file_format
      <td>Identification of the genomic file format of the initial input file (eg. fastq, bam, cram).
      <td><a href="http://edamontology.org/format_1921">EDAM’s file types and formats</a>
      <td>1..1
    <tbody>
     <tr>
      <td>Final_output_file_format
      <td>Identification of the genomic file format of the final output file (eg. vcf, gvcf).
      <td><a href="http://edamontology.org/format_1921">EDAM’s file types and formats</a>
      <td>1..1
    <tbody>
     <tr>
      <td>Final_output_file_format_version
      <td>Identification of the version of genomic file format of the final output file (eg. vcf, gvcf).
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Alignment_software
      <td>Identification of the software used for alignment.
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Alignment_Genome
      <td>The specific build of the human genome used as reference for sequence alignment and variant calling.
      <td>--> Digital Resource class
      <td>1..n
    <tbody>
     <tr>
      <td>Specific_Settings_Alignment_Genome
      <td> Any specific settings regarding alternative contigs or decoys.
      <td>String
      <td>0..n
    <tbody>
     <tr>
      <td>Target Gene
      <td>In case of targeted sequencing, specify which gene is being targeted. This item points to another class: Target_Gene.
      <td>--> Target Gene class
      <td>0..n
    <tbody>
     <tr>
      <td>Target Other
      <td>Any other targeted genomic region.
      <td>Children of <a href="http://www.sequenceontology.org/browser/current_release/term/SO:0000001">SIO region</a> <br> NULL flavors
      <td>0..n
    <tbody>
     <tr>
      <td>Panel_of_Normals_Included
      <td>Indicate whether a panel of normals is included during variant calling.
      <td>Boolean
      <td>0..1
    <tbody>
     <tr>
      <td>Panel_of_Normals_Description
      <td>Free text description of panel of normals, if applicable.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Variant
      <td>A detected and reported variant.
      <td>--> Variant class
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_calling
      <td>Identification of the software used for variant calling.
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_calling_date
      <td>Defines the date of variant calling.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
    <tbody>
     <tr>
      <td>Variant_Annotation
      <td>Identification of the software used for variant annotation.
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_Annotation_database
      <td>Database and version used for variant annotation.
      <td>--> Digital Resource class
      <td>0..n
   </table>
   <h4 class="heading settled" data-level="2.4.1" id="digital-resource"><span class="secno">2.4.1. </span><span class="content">Digital Resource</span><a class="self-link" href="#digital-resource"></a></h4>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Name
      <td>The name of the tool/software/database used.
      <td>String
      <td>1..1
    <tbody>
     <tr>
      <td>Website
      <td>Link to the website or repository (like GitHub) of the tool/software/database.
      <td>URL
      <td>0..n
    <tbody>
     <tr>
      <td>Identifier
      <td>bio.tools identifier for the digital resource.
      <td><a href="https://bio.tools/">bio.tools</a> identifier
      <td>0..1
    <tbody>
     <tr>
      <td>Version
      <td>The version of the tool/software/database used.
      <td>double
      <td>1..1
    <tbody>
     <tr>
      <td>Date used
      <td>The date when the tool/software/database was last used.
      <td>xsd:dateTime
      <td>1..1
    <tbody>
     <tr>
      <td>Settings
      <td>Free text account of the settings used in the tool/software/database.
      <td>String
      <td>0..n
    <tbody>
     <tr>
      <td>Parameters
      <td>Description of parameters used with the specified software. Copy the complete command line (all lines executed) used.
      <td>String
      <td>0..n
   </table>
   <h4 class="heading settled" data-level="2.4.2" id="variant"><span class="secno">2.4.2. </span><span class="content">Variant</span><a class="self-link" href="#variant"></a></h4>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Variant_Type
      <td>The category or type of variation or abnormality present in an amino acid or nucleic acid sequence.
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C164674">SNVs</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C99752">indels</a>, SVs, CNVs, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C20195">gene fusions</a>, ... (to be extended).
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_Origin
      <td>A quality inhering in a variant by virtue of its origin.
      <td><a href="http://purl.obolibrary.org/obo/SO_0001777">somatic</a>, <a href="http://purl.obolibrary.org/obo/SO_0001778">germline</a>, <a href="http://purl.obolibrary.org/obo/SO_0001775">maternal</a>, <a href="http://purl.obolibrary.org/obo/SO_0001776">paternal</a>, <a href="http://purl.obolibrary.org/obo/SO_0001779">pedigree specific</a>, <a href="http://purl.obolibrary.org/obo/SO_0001780">population specific</a>, <a href="http://purl.obolibrary.org/obo/SO_0001781">de novo</a>.
      <td>0..1
    <tbody>
     <tr>
      <td>Variant_representation
      <td>The representation of the variant using HGVS nomenclature.
      <td>String following <a href="https://hgvs-nomenclature.org/stable/">HGVS nomenclature</a>
      <td>1..1
    <tbody>
     <tr>
      <td>Clinical_Variant_Interpretation_criteria
      <td>Internationally (e.g. ACMG, ESMO-ESCAT) criteria met for variant interpretation
      <td>List of versions of ACMG, ESMO-ESCAT, others??
      <td>0..n
    <tbody>
     <tr>
      <td>Clinical_Variant_Interpretation_result
      <td>Indicates result of clinical variant interpretation.
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C168802">benign</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C168801">likely benign</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C94187">VUS</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C168800">likely pathogenic</a>, <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C168799">pathogenic</a>.
      <td>0..n
    <tbody>
     <tr>
      <td>Clinical_expert_panel_decision
      <td>Decision by clinical expert panel concerning the variant interpretation
      <td>String
      <td>0..n
    <tbody>
     <tr>
      <td>Applied_Criteria_of_Evidence
      <td>A category which fits with categories provided by Expert panels or tools accepted in Clincial Practice. If such recommendations are not available the weighted categories provided by freely available tools would be acceptable.
      <td>As listed in tables 3 and 4 in <a href="https://doi.org/10.1038/gim.2015.30">Standards and guidelines for the interpretation of sequence variants: a joint consensus recommendation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology</a>
      <td>0..n
    <tbody>
     <tr>
      <td>Clinical_Interpretation_Tool
      <td>Identification of the tool used for clinical interpretation
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_calling_software_deviation
      <td>Identification of the software used for variant calling, if different from software stated in Sequence class.
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_Annotation_tools_deviation
      <td>Identification of the software used for variant annotation, if different from software stated in Sequence class. 
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Variant_Annotation_database_deviation
      <td>Database and version used for variant annotation, if different from software stated in Sequence class.
      <td>--> Digital Resource class
      <td>0..n
    <tbody>
     <tr>
      <td>Reported_to_patient
      <td>Indication if the variant has been reported back to the patient, if different from software stated in Sequence class.
      <td>Boolean
      <td>0..1
   </table>
   <h4 class="heading settled" data-level="2.4.3" id="target-gene"><span class="secno">2.4.3. </span><span class="content">Target Gene</span><a class="self-link" href="#target-gene"></a></h4>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>URI
      <td>URI identifying the targeted gene.
      <td>URI to either <a href="https://www.genenames.org/">HGNC</a> <a href="https://www.ncbi.nlm.nih.gov/gene">NCBI gene</a>, <a href="https://www.omim.org/">OMIM</a>, <a href="https://hpo.jax.org/">HPO</a> or <a href="https://hgvs-nomenclature.org/stable/">HGVS</a> for variants.
      <td>0..1
    <tbody>
     <tr>
      <td>Label
      <td>Label of the target gene, if no URI can be provided.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Description
      <td>Description of target gene.
      <td>String
      <td>0..1
   </table>
   <h3 class="heading settled" data-level="2.5" id="treatment"><span class="secno">2.5. </span><span class="content">Treatment</span><a class="self-link" href="#treatment"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Intention_to_Treat
      <td>Indicate the intended disease outcome for which the treatment is given, may be coded as SNOMED-CT.
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/373808002">SCTID: 373808002</a> | Curative - procedure intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/373847000">SCTID: 373847000</a> | Neoadjuvant intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/373846009">SCTID: 373846009</a> | Adjuvant - intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/363676003">SCTID: 363676003</a> | Palliative - procedure intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/360271000">SCTID: 360271000</a> | Prophylaxis - procedure intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/243114000">SCTID: 243114000</a> | Support (regime/therapy)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/129304002">SCTID: 129304002</a> | Excision - action 
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/261004008">SCTID: 261004008</a> | Diagnostic intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/129428001">SCTID: 129428001</a> | Preventive - intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/429892002">SCTID: 429892002</a> | Guidance intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/360156006">SCTID: 360156006</a> | Screening - procedure intent
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/447295008">SCTID: 447295008</a> | Forensic intent
            <br>Other - HL7 NULL flavor (OTH, UNK, NI, NA)
      <td>0..1
    <tbody>
     <tr>
      <td>Setting
      <td>Indicate the treatment setting, which describes the treatment’s purpose in relation to the primary treatment (Reference: NCIT C124308)
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C15675">Adjuvant</a> <br> <a href="https://w3id.org/fair-genomes/resource/FG_0000760">Advanced/Metastatic</a><br> <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C15665">Neoadjuvant</a><br> <a href="https://terminology.hl7.org/5.1.0/CodeSystem-v3-NullFlavor.html#v3-NullFlavor-NA">Not applicable</a>
      <td>0..1
    <tbody>
     <tr>
      <td>Type
      <td>Indicates the type of treatment regimen that the patient completed; coded values can be chosen, SNOMED may be chosen for "procedures".
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/23719005">SCTID: 23719005</a> | Bone marrow transplant
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/367336001">SCTID: 367336001</a> | Chemotherapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/18629005">SCTID: 18629005</a> | Medication
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/423827005">SCTID: 423827005</a> | Endoscopic therapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/169413002">SCTID: 169413002</a> | Hormonal therapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/76334006">SCTID: 76334006</a> | Immunotherapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/257891001">SCTID: 257891001</a> | Photodynamic therapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1287742003">SCTID: 1287742003</a> | Radiation therapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/56305001">SCTID: 56305001</a> | Radionucleotide therapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1269349006">SCTID: 1269349006</a> | Stem cell transplant
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/387713003">SCTID: 387713003</a> | Surgery
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/394613000">SCTID: 394613000</a> | Gene therapy
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/310341009">SCTID: 310341009</a> or 373818007 | Watchful waiting
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/424313000">SCTID: 424313000</a> | Active follow-up
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/703423002">SCTID: 703423002</a> | Chemoradiotherapy
            <br>HL7 NULL flavor (NI, NA, OTH) | Other
      <td>1..1
    <tbody>
     <tr>
      <td>Subtype
      <td>Detailed specification of the treatment type.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Line
      <td>Indicate if the treatment was the primary treatment following the initial diagnosis or 2nd, 3rd, …
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>treatmentCode
      <td>A drug product that contains one or more active and/or inactive ingredients used by the patient intended to treat, prevent or alleviate the symptoms of disease. Any hormone therapies, gender-related or otherwise, should also be recorded here.
      <td><a href="https://raw.githubusercontent.com/fairgenomes/fairgenomes-semantic-model/main/lookups/Drugs.txt">ATC codes</a>, IDMP when available. <br> In case of Cancer: <a href="https://hemonc.org/wiki/Main_Page">HEMONC</a>, <a href="https://mor.nlm.nih.gov/RxClass/">RxNorm</a>
      <td>1..n
    <tbody>
     <tr>
      <td>Date_start_overall
      <td>The date and time of (the start of) the treatment.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
    <tbody>
     <tr>
      <td>Date_end_overall
      <td>The date and time of the end of the overall treatment.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
    <tbody>
     <tr>
      <td>Intended Duration
      <td>The duration of treatment regimen, in days.
      <td>Integer
      <td>0..1
    <tbody>
     <tr>
      <td>DoseUnits
      <td>Indicates the total dose given in units (e.g. of Gray (Gy) when radiation, Millimeters/24hours for medication).
      <td>Choicelist with most common DoseUnits, like Gray, Milligrams/24 hours: make use of <a href="https://ucum.org/">UCUM</a>.
      <td>0..1
    <tbody>
     <tr>
      <td>cumulativeDose
      <td>The amount of any substance administered over a specific period of time.
      <td>Float
      <td>0..1
    <tbody>
     <tr>
      <td>doseIntervals
      <td>The number of times a substance is administered within a specific time period.
      <td>ISO8601 Period
      <td>0..1
    <tbody>
     <tr>
      <td>routeOfAdministration
      <td>Designation of the part of the body through which or into which, or the way in which, the medicinal product is intended to be introduced. In some cases a medicinal product can be intended for more than one route and/or method of administration.
      <td>Subclass of <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C13442">Anatomy qualifier</a>
      <td>0..n
    <tbody>
     <tr>
      <td>Pre-Treated
      <td>Indicates if disease has been pre-treated or is in course of treatment.
      <td>Boolean
      <td>0..1
    <tbody>
     <tr>
      <td>Procedures
      <td>Indicates the type of treatment regimen that the patient completed; coded values can be chosen, SNOMED may be chosen for "procedures".
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/71388002">SNOMED Procedure</a>
      <td>0..1
    <tbody>
     <tr>
      <td>Surgical resection quality
      <td>Evaluation of the surgical resection quality based on coded values.
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222638005">SCTID: 1222638005</a> |American Joint Committee on Cancer R0 (qualifier value)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222639002">SCTID: 1222639002</a> | American Joint Committee on Cancer R1 (qualifier value)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222640000">SCTID: 1222640000</a> | American Joint Committee on Cancer R2 (qualifier value)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1222641001">SCTID: 1222641001</a> | American Joint Committee on Cancer RX (qualifier value)
      <td>0..1
    <tbody>
     <tr>
      <td>Treatment_Status
      <td>Indicates the patient’s outcome of the prescribed treatment (coded values, e.g. treatment not completed | because of toxicity).
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/182992009">SCTID: 182992009</a> | Treatment completed (situation)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/445528004">SCTID: 445528004</a> | Treatment changed (situation)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/405613005">SCTID: 405613005</a> | Planned procedure (situation)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/416406003">SCTID: 416406003</a> | Procedure discontinued (situation)
      <td>0..1
    <tbody>
     <tr>
      <td>Reason_for_incomplete_Treatment
      <td>Reason for discontinuation of the treatment.
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/419099009">SCTID: 419099009</a> | Dead (finding)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/1296859006">SCTID: 1296859006</a> | Procedure declined (situation)   
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/713247000">SCTID: 713247000</a> | Procedure discontinued by patient (situation)
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/713246009">SCTID: 713246009</a> | Procedure discontinued by healthcare professional (situation) 
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/266721009">SCTID: 266721009</a> | Absent response to treatment (situation)       
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/407563006">SCTID: 407563006</a> | Treatment not tolerated (situation) 
            <br>technical or organizational problems | to be added, or use OTH 
            <br>Other | HL7 null flavour OTH      
            <br>Not applicable | HL7 null flavour, NA
            <br>No information | HL7 null flavour, NI
      <td>0..1
    <tbody>
     <tr>
      <td>Response_to_Treatment
      <td>The patients' response to the applied treatment regimen. (Source: RECIST).
      <td><a href="https://w3id.org/fair-genomes/resource/FG_0000761">Complete Response</a> <br>
            <a href="https://w3id.org/fair-genomes/resource/FG_0000762">Disease progression</a> <br>
            <a href="https://w3id.org/fair-genomes/resource/FG_0000763">NED</a> <br>
            <a href="https://w3id.org/fair-genomes/resource/FG_0000764">Partial Response</a> <br>
            <a href="https://w3id.org/fair-genomes/resource/FG_0000765">Stable Disease</a>
      <td>0..1
    <tbody>
     <tr>
      <td>Adverse_Events
      <td>Reports any treatment related adverse events. (Codelist reference: NCI-CTCAE (v5.0))
      <td><a href="https://ctep.cancer.gov/protocoldevelopment/electronic_applications/docs/CTCAE_v5_Quick_Reference_5x7.pdf">CTCAE Codes</a>
      <td>0..n
    <tbody>
     <tr>
      <td>Toxicity_Type
      <td>If the treatment was terminated early due to acute toxicity, indicates the type of toxicity that caused early termination of treatment.
      <td>Children of <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C27990">Toxicity (NCIT)</a>
      <td>0..1
    <tbody>
     <tr>
      <td>Modality
      <td>Indicates the method of radiation treatment or modality.
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C15650">Electron</a> <br> <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C15458">Heavy Ions</a> <br> <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C93337">Photon</a> <br> <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C66897">Proton</a>
      <td>0..n
    <tbody>
     <tr>
      <td>Fractions
      <td>Indicates the total number of fractions delivered as part of radiation treatment.
      <td>Integer
      <td>0..1
    <tbody>
     <tr>
      <td>Site
      <td>Indicates the body region where radiation therapy was administered.
      <td>Children concepts of <a href="http://purl.bioontology.org/ontology/SNOMEDCT/91723000">Anatomical Structure (body structure)</a>
      <td>0..n
   </table>
   <h3 class="heading settled" data-level="2.6" id="biomarker"><span class="secno">2.6. </span><span class="content">Biomarker</span><a class="self-link" href="#biomarker"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Purpose
      <td>Necessary information to indicate the objective of the biomarker used  (coded value, e.g. diagnostic, prognostic,…).
      <td><a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C201361">Diagnosis</a> <br>
            <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C201363">Prognosis</a> <br>
            <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C201364">Prediction</a> <br>
            <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C201362">Monitoring</a>
      <td>0..1
    <tbody>
     <tr>
      <td>Type
      <td>What type is the biomarker classified as.
      <td>Molecular
            <br>Imaging
            <br>Anthropometric
            <br>Cellular
            <br>Physiological
      <td>1..1
    <tbody>
     <tr>
      <td>Subtype
      <td>Subtype of the biomarker as classified in the type.
      <td>If Type ==
            <br>- <b>Molecular</b>: Genetics/Genomics,
            Epigenetics/Epigenomics,
            Transcription/Transcriptomics,
            Metabolites/Metabolomics,
            Proteins/Proteomics,
            Microbiomics/Microbiology,
            Biochemistry (biochemical),
            Other molecular biomarker,
            N/P
            <br>- <b>Imaging</b>: X-Rays,
            Ultrasound (echography, etc),
            CT Scan,
            PET/SPECT,
            Spectrometry,
            MRI,
            Scintigraphy (Gamma),
            Mammography,
            Other image biomarker,
            N/P
            <br>- <b>Anthropometric</b>: BMI,
            Body perimeters (circumference),
            Other anthropometric biomarker,
            N/P
            <br>- <b>Cellular</b>: Histology (tissue abnormalities),
            Cytology (cell types),
            Other cellular biomarker,
            N/P
            <br>- <b>Physiological</b>: Blood Pressure,
            Ankle-brachial Index,
            ECG,
            EEG,
            Electromyography,
            Other physiological biomarker,
            N/P
      <td>1..1
    <tbody>
     <tr>
      <td>Name
      <td>Biomarker name.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Code
      <td>Code of the biomarker.
      <td>String
      <td>0..1
    <tbody>
     <tr>
      <td>Code System
      <td>Code System of the Biomarker Code.
      <td>String or URI
      <td>0..1
    <tbody>
     <tr>
      <td>Date
      <td>The date of the biomarker being obtained.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..1
   </table>
   <h3 class="heading settled" data-level="2.7" id="environmental-exposure"><span class="secno">2.7. </span><span class="content">Environmental Exposure</span><a class="self-link" href="#environmental-exposure"></a></h3>
   <table>
    <thead>
     <tr>
      <th>Item
      <th>Definition
      <th>Value
      <th>Cardinality
    <tbody>
     <tr>
      <td>Tobacco Use Status
      <td>The status of the patient’s tobacco use.
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/449868002">SCTID:449868002</a> | Smokes tobacco daily
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/428041000124106">SCTID:428041000124106</a> | Occasional tobacco smoker
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/43381005">SCTID:43381005</a> | Passive smoker
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/8517006">SCTID:8517006</a> | Ex-smoker
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/405746006">SCTID:405746006</a> | Current non smoker but past smoking history unknown
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/266919005">SCTID:266919005</a> | Never smoked tobacco
            <br>NullFlavor <a href="https://terminology.hl7.org/1.0.0/CodeSystem-v3-NullFlavor.html#v3-NullFlavor-OTH">OTH</a> 
      <td>0..1
    <tbody>
     <tr>
      <td>Type Of Tobacco Used
      <td>Type of tobacco the patient uses.
      <td>Children of <a href="https://addictovocab.org/ADDICTO_0000311">tobacco product</a> combined with childred of <a href="https://addictovocab.org/ADDICTO_0000316">vaporizer</a>, or
            <br> <a href="https://terminology.hl7.org/1.0.0/CodeSystem-v3-NullFlavor.html">HL7 NULL flavors</a>
      <td>1..n (conditional)
    <tbody>
     <tr>
      <td>Smoking amount
      <td>The number of cigarettes, cigars or grams of rolling tobacco consumed per day, week, month or year.
      <td>amount/day | amount/week | amount/month | amount/year
      <td>1..1 (conditional)
    <tbody>
     <tr>
      <td>Pack years
      <td>The unit indicating the smoker’s total exposure to tobacco smoke. For cigarettes, this is calculated using the number of smoked packs of cigarettes per day (one pack = 20 cigarettes) times the number of years of smoking. For other forms of tobacco, this is usually converted to an equivalent cigarette consumption. Often, only the number of pack years is estimated.
      <td>Integer 
      <td>1..1 (conditional)
    <tbody>
     <tr>
      <td>Alcohol use status
      <td>The status of the patient’s alcohol use.
      <td><a href="http://purl.bioontology.org/ontology/SNOMEDCT/219006">SCTID:219006</a> | Current drinker
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/105542008">SCTID:105542008</a> | Non - drinker
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/82581004">SCTID:82581004</a> | Ex-drinker
            <br><a href="http://purl.bioontology.org/ontology/SNOMEDCT/783261004">SCTID:783261004</a> | Lifetime non-drinker of alcohol
            <br>NullFlavor <a href="https://terminology.hl7.org/1.0.0/CodeSystem-v3-NullFlavor.html#v3-NullFlavor-OTH">OTH</a> 
      <td>0..1
    <tbody>
     <tr>
      <td>Alcohol amount
      <td>The extent of the patient’s alcohol use in units of alcohol per time period (day/week/year).
      <td>AU per (day/week/year). 1 A.U.= 
            <br>125 ml of wine, 
            <br>330 ml of beer, 
            <br>80 ml of drink, 
            <br>40 ml liquor
      <td>0..n
    <tbody>
     <tr>
      <td>Other Type of Exposure
      <td>Indicates other exposures other than tabacco/smoking and alcohol use, for example asbestos. This exposure can include direct physical contact, inhalation, ingestion, or residing in close proximity to the source.
      <td>- Inadequate water (quantity and quality), sanitation and solid waste disposal, improper hygein (handwashing)
            <br>- Improper water resource managment, uncluding poor drainage,
            <br>- Crowded housing and poor ventialtion of smoke
            <br>- Exposures to vehicular and industrial air pollution
            <br>- Population movement and encroachment and construction, which affect feeding and breeding grounds of vectors, such as mosquitoes,
            <br>- Exposure to naturally ocurring toxic substances
            <br>- Natural resources degradation (for example, landslides, poor drainage, erosion)
            <br>- Climate change, partly from combustion of fossil fuel and release of greenhouse gases in transportation, industry, and poor energy conservation in housing, fuel, commerce, and industry
            <br>- Ozone depletion from industrial and commercial actvitiy
            <br><b>Biological hazard</b> (Bacteria, Fungi, Parasitic worms, Protozoa, Viruses, Prions)
            <br><b>Chemical hazards</b> (Air pollutant, Heavy metal (Arsenice, Mercury, Lead), Pesticides, Other ( Formaldehyde, Asbestos, PFAS, PCBs, BPA, Phthalates, Radon, DDT)
            <br><b>Physical environmental hazard</b> (Radiation, Natural disaster (narrow matches: exposure to earthquake, exposure to flooding, exposure to tsunami etc.), Extreme weather, Human activities (eg. traffic accident))
            <br><a href="https://terminology.hl7.org/1.0.0/CodeSystem-v3-NullFlavor.html">HL7 NULL flavors</a>
      <td>0..n
    <tbody>
     <tr>
      <td>Other_Other Type of Exposure
      <td>Provide the option to if "Other" option is choosen from List of other exposures, type a free text field.
      <td>String
      <td>1..1 (conditional)
    <tbody>
     <tr>
      <td>Travel History
      <td>Description of any travel within 4 weeks before the diagnosis.
      <td>String
      <td>0..n
    <tbody>
     <tr>
      <td>Travel History_start_date
      <td>Start date of relevant travel history.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..n
    <tbody>
     <tr>
      <td>Travel History_end_date
      <td>End date of relevant travel history.
      <td>Date (YYYY-MM-DD), ISO 8601 format
      <td>0..n
    <tbody>
     <tr>
      <td>Residential area at risk
      <td>Description of the area of ​​known environmental exposure conditions where the subject/patient resides.
      <td>String
      <td>0..n
   </table>
   <h2 class="heading settled" data-level="3" id="matches"><span class="secno">3. </span><span class="content">Matches</span><a class="self-link" href="#matches"></a></h2>
   <p>Different countries use different standards or vocabularies for their data. For example, in the harmonized minimal data model we mostly use terms from SNOMED CT and NCIT, but some countries only use ICD-10, and it’s difficult for them to convert everything to SNOMED.
To make things easier, we decided to focus on matches between terms from different systems. This way, a dataset using ICD-10 can still be included in the GDI Catalogue, as long as we can clearly show how its terms connect to SNOMED or other terms.</p>
   <p>For example:</p>
   <ul>
    <li data-md>
     <p>One dataset might use a SNOMED term to describe a disease.</p>
    <li data-md>
     <p>Another dataset might describe the same disease using an ICD-10 term.</p>
    <li data-md>
     <p>Without a match between them, our system would think they are two different things.</p>
    <li data-md>
     <p>If we add an exact match, the system will understand that they are the same, just described with different vocabularies.</p>
   </ul>
   <p>This will make data both human and machine readable.</p>
   <h3 class="heading settled" data-level="3.1" id="how-to-use-matches"><span class="secno">3.1. </span><span class="content">How To Use Matches</span><a class="self-link" href="#how-to-use-matches"></a></h3>
   <p>We started by adding different types of matches to all items and values in the model. The types of matches (match attribute), their definitions and examples can be found in the table below.</p>
   <table>
    <thead>
     <tr>
      <th>Match Attribute
      <th>Definition
      <th>Example
    <tbody>
     <tr>
      <td><a href="https://www.w3.org/2009/08/skos-reference/skos.html#exactMatch">skos:exactMatch</a>
      <td>Two concepts have the same meaning and can be used interchangeably in all contexts. This link is transitive, meaning if A is an exact match to B, and B is an exact match to C, then A is also an exact match to C.
      <td><a href="https://browser.ihtsdotools.org/?perspective=full&amp;conceptId1=184099003&amp;edition=MAIN&amp;release=&amp;languages=en">SNOMED Date of birth</a> has an exact match to <a href="http://www.ebi.ac.uk/efo/EFO_0004950">EFO Date of birth</a>. Both refer to the same concept with no variation in meaning.
     <tr>
      <td><a href="https://www.w3.org/2009/08/skos-reference/skos.html#closeMatch">skos:closeMatch</a>
      <td>Two concepts are sufficiently similar that they can be used interchangeably in many applications and schemes, but they are not strictly identical. This link is not meant to be transitive.
      <td><a href="https://browser.ihtsdotools.org/?perspective=full&amp;conceptId1=116154003&amp;edition=MAIN&amp;release=&amp;languages=en">SNOMED Patient</a> has a close match to <a href="https://evsexplore.semantics.cancer.gov/evsexplore/concept/ncit/C142710">NCIT Study Participant</a>. <br> A patient is an individual receiving medical care, while a study participant is someone enrolled in a research study. In some cases, these terms can be used interchangeably (e.g., in clinical trials involving patients), but not all study participants are patients—some might be healthy controls.
     <tr>
      <td><a href="https://www.w3.org/2009/08/skos-reference/skos.html#relatedMatch">skos:relatedMatch</a>
      <td>Two concepts are associated but not similar enough to be considered exact or close matches.
      <td>   The concept <a href="https://identifiers.org/hgnc:1100">HGNC BRCA1</a> and 
            <a href="http://purl.obolibrary.org/obo/NCIT_C4872">NCIT Breast Carcinoma</a> 
            have a skos:relatedMatch relationship because they are strongly associated but not identical. 
            BRCA1 is a gene mutation, while breast cancer is a disease that may develop as a result of certain BRCA1 mutations. 
            However, not all BRCA1 mutations lead to breast cancer, and not all cases of breast cancer are caused by BRCA1 mutations, making them related but not broader/narrower concepts.
     <tr>
      <td><a href="https://www.w3.org/2009/08/skos-reference/skos.html#broadMatch">skos:broadMatch</a>
      <td>The concept represented by the external entity is broader than the item. This is a reverse property of skos:narrowMatch.
      <td>The concept <a href="https://ontobee.org/ontology/NCIT?iri=http://purl.obolibrary.org/obo/NCIT_C4872">NCIT Breast Carcinoma</a> has a skos:broadMatch relationship to <a href="https://browser.ihtsdotools.org/?perspective=full&amp;conceptId1=363346000&amp;edition=MAIN&amp;release=&amp;languages=en">SNOMED Cancer</a>. This means that Cancer is the broader concept because it represents a general category that includes multiple types of cancer, one of which is Breast Cancer.
     <tr>
      <td><a href="https://www.w3.org/2009/08/skos-reference/skos.html#narrowMatch">skos:narrowMatch</a>
      <td>The concept represented by the external entity is narrower than the item. This is a reverse property of skos:broadMatch.
      <td>The concept <a href="https://browser.ihtsdotools.org/?perspective=full&amp;conceptId1=363346000&amp;edition=MAIN&amp;release=&amp;languages=en">SNOMED Cancer</a> has a skos:narrowMatch relationship to <a href="https://ontobee.org/ontology/NCIT?iri=http://purl.obolibrary.org/obo/NCIT_C4872">NCIT Breast Carcinoma</a>. This means that Breast Cancer is a specific subtype of Cancer.
   </table>
   <p>Next to the matches, we also record what is the source of the match. If the mapping has already been done, for example SNOMED to ICD-10 in the SNOMED browser, or in the T-Rex browser, we include this information with the match in the spreadsheet.</p>
   <h3 class="heading settled" data-level="3.2" id="using-sssom"><span class="secno">3.2. </span><span class="content">Using SSSOM</span><a class="self-link" href="#using-sssom"></a></h3>
   
From now on, we decided to follow the <a href="https://mapping-commons.github.io/sssom/">SSSOM (Simple Standard for Sharing Ontological Mappings)</a> method. This is a standard format for describing matches between terms. We will:

   <ul>
    <li data-md>
     <p>Add all required <a href="https://mapping-commons.github.io/sssom/linkml-index/">SSSOM elements</a>.</p>
    <li data-md>
     <p>Include recommended and optional elements if they are useful.</p>
    <li data-md>
     <p>Make sure our matches follow the SSSOM rules.</p>
   </ul>
   <p>Using this model and the SSSOM standard will help us:</p>
   <ul>
    <li data-md>
     <p>Make it easier for teams using different vocabularies to share their data.</p>
    <li data-md>
     <p>Let people and software search and analyze the data more effectively.</p>
    <li data-md>
     <p>Support more countries and teams to join the GDI Catalogue without needing to change everything.</p>
   </ul>
  </main>
  <div data-fill-with="conformance">
   <h2 class="heading no-num no-ref settled" id="conformance"><span class="content">
    Conformance</span><a class="self-link" href="#conformance"></a></h2>
   <p>
            Conformance requirements are expressed with a combination of descriptive assertions and RFC 2119 terminology.
            The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL”
            in the normative parts of this document
            are to be interpreted as described in RFC 2119.
            However, for readability,
            these words do not appear in all uppercase letters in this specification.

        </p>
   <p>
            All of the text of this specification is normative
            except sections explicitly marked as non-normative, examples, and notes. <a data-link-type="biblio" href="#biblio-rfc2119" title="Key words for use in RFCs to Indicate Requirement Levels">[RFC2119]</a>

        </p>
   <p>
            Examples in this specification are introduced with the words “for example”
            or are set apart from the normative text with <code>class="example"</code>, like this:

        </p>
   <div class="example" id="example-example"><a class="self-link" href="#example-example"></a>
            This is an example of an informative example.
        </div>
   <p>
            Informative notes begin with the word “Note”
            and are set apart from the normative text with <code>class="note"</code>, like this:

        </p>
   <p class="note" role="note">
            Note, this is an informative note.</p>
  </div>
<script>
(function() {
  "use strict";
  var collapseSidebarText = '<span aria-hidden="true">←</span> '
                          + '<span>Collapse Sidebar</span>';
  var expandSidebarText   = '<span aria-hidden="true">→</span> '
                          + '<span>Pop Out Sidebar</span>';
  var tocJumpText         = '<span aria-hidden="true">↑</span> '
                          + '<span>Jump to Table of Contents</span>';

  var sidebarMedia = window.matchMedia('screen and (min-width: 78em)');
  var autoToggle   = function(e){ toggleSidebar(e.matches) };
  if(sidebarMedia.addListener) {
    sidebarMedia.addListener(autoToggle);
  }

  function toggleSidebar(on) {
    if (on == undefined) {
      on = !document.body.classList.contains('toc-sidebar');
    }

    /* Don't scroll to compensate for the ToC if we're above it already. */
    var headY = 0;
    var head = document.querySelector('.head');
    if (head) {
      // terrible approx of "top of ToC"
      headY += head.offsetTop + head.offsetHeight;
    }
    var skipScroll = window.scrollY < headY;

    var toggle = document.getElementById('toc-toggle');
    var tocNav = document.getElementById('toc');
    if (on) {
      var tocHeight = tocNav.offsetHeight;
      document.body.classList.add('toc-sidebar');
      document.body.classList.remove('toc-inline');
      toggle.innerHTML = collapseSidebarText;
      if (!skipScroll) {
        window.scrollBy(0, 0 - tocHeight);
      }
      tocNav.focus();
      sidebarMedia.addListener(autoToggle); // auto-collapse when out of room
    }
    else {
      document.body.classList.add('toc-inline');
      document.body.classList.remove('toc-sidebar');
      toggle.innerHTML = expandSidebarText;
      if (!skipScroll) {
        window.scrollBy(0, tocNav.offsetHeight);
      }
      if (toggle.matches(':hover')) {
        /* Unfocus button when not using keyboard navigation,
           because I don't know where else to send the focus. */
        toggle.blur();
      }
    }
  }

  function createSidebarToggle() {
    /* Create the sidebar toggle in JS; it shouldn't exist when JS is off. */
    var toggle = document.createElement('a');
      /* This should probably be a button, but appearance isn't standards-track.*/
    toggle.id = 'toc-toggle';
    toggle.class = 'toc-toggle';
    toggle.href = '#toc';
    toggle.innerHTML = collapseSidebarText;

    sidebarMedia.addListener(autoToggle);
    var toggler = function(e) {
      e.preventDefault();
      sidebarMedia.removeListener(autoToggle); // persist explicit off states
      toggleSidebar();
      return false;
    }
    toggle.addEventListener('click', toggler, false);


    /* Get <nav id=toc-nav>, or make it if we don't have one. */
    var tocNav = document.getElementById('toc-nav');
    if (!tocNav) {
      tocNav = document.createElement('p');
      tocNav.id = 'toc-nav';
      /* Prepend for better keyboard navigation */
      document.body.insertBefore(tocNav, document.body.firstChild);
    }
    /* While we're at it, make sure we have a Jump to Toc link. */
    var tocJump = document.getElementById('toc-jump');
    if (!tocJump) {
      tocJump = document.createElement('a');
      tocJump.id = 'toc-jump';
      tocJump.href = '#toc';
      tocJump.innerHTML = tocJumpText;
      tocNav.appendChild(tocJump);
    }

    tocNav.appendChild(toggle);
  }

  var toc = document.getElementById('toc');
  if (toc) {
    createSidebarToggle();
    toggleSidebar(sidebarMedia.matches);

    /* If the sidebar has been manually opened and is currently overlaying the text
       (window too small for the MQ to add the margin to body),
       then auto-close the sidebar once you click on something in there. */
    toc.addEventListener('click', function(e) {
      if(e.target.tagName.toLowerCase() == "a" && document.body.classList.contains('toc-sidebar') && !sidebarMedia.matches) {
        toggleSidebar(false);
      }
    }, false);
  }
  else {
    console.warn("Can't find Table of Contents. Please use <nav id='toc'> around the ToC.");
  }

  /* Wrap tables in case they overflow */
  var tables = document.querySelectorAll(':not(.overlarge) > table.data, :not(.overlarge) > table.index');
  var numTables = tables.length;
  for (var i = 0; i < numTables; i++) {
    var table = tables[i];
    var wrapper = document.createElement('div');
    wrapper.className = 'overlarge';
    table.parentNode.insertBefore(wrapper, table);
    wrapper.appendChild(table);
  }

})();
</script>
  <h2 class="heading no-num no-ref settled" id="references"><span class="content">References</span><a class="self-link" href="#references"></a></h2>
  <h3 class="heading no-num no-ref settled" id="normative"><span class="content">Normative References</span><a class="self-link" href="#normative"></a></h3>
  <dl>
   <dt id="biblio-rfc2119">[RFC2119]
   <dd>S. Bradner. <a href="https://datatracker.ietf.org/doc/html/rfc2119"><cite>Key words for use in RFCs to Indicate Requirement Levels</cite></a>. March 1997. Best Current Practice. URL: <a href="https://datatracker.ietf.org/doc/html/rfc2119">https://datatracker.ietf.org/doc/html/rfc2119</a>
  </dl>